{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "######################################################\n",
    "import csv\n",
    "import pickle\n",
    "from copy import deepcopy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def readcounts_to_rpm(readcounts):\n",
    "    for chrom in readcounts.keys():\n",
    "        totalread = sum(readcounts[chrom])\n",
    "        for position in range(len(readcounts[chrom])):\n",
    "            readcounts[chrom][position]/=float(totalread)\n",
    "            readcounts[chrom][position]*=1E6\n",
    "            \n",
    "def writecountsf(rpm,filepath): \n",
    "    import struct\n",
    "    f2=open(filepath+\"keys\",\"w\")\n",
    "    for chrom in rpm.keys():\n",
    "        f=open(filepath+\"genome\",\"wb\")\n",
    "        for position in rpm[chrom]:\n",
    "            f.write(struct.pack(\"f\",position))\n",
    "        f.close()\n",
    "        f2.write(chrom+\"\\n\")\n",
    "    f2.close()\n",
    "\n",
    "def countstowig(rpm,filestring):\n",
    "    import random\n",
    "    f=open(filestring+\".wig\",\"w\")\n",
    "    filestring=filestring.partition(\"_\")[0][-3:]\n",
    "\n",
    "    random.seed(filestring)\n",
    "    c1=random.randint(0,255)\n",
    "    random.seed(filestring+\"xxx\")\n",
    "    c2=random.randint(0,255)\n",
    "    random.seed(filestring+\"000\")\n",
    "    c3=random.randint(0,255)\n",
    "\n",
    "    f.write(\"track name=tracklabel viewLimits=-5:5 color=\"+str(c1)+','+str(c2)+','+str(c3)+\"\\n\")\n",
    "    for chrom in rpm.keys():\n",
    "        if chrom[0:3]=='chr':\n",
    "            f.write(\"fixedStep  chrom=\"+chrom+\"  start=1  step=1\\n\")\n",
    "        else:\n",
    "            f.write(\"fixedStep  chrom=\\\"\"+chrom+\"\\\"  start=1  step=1\\n\") \n",
    "\n",
    "\n",
    "        for position in rpm[chrom]:\n",
    "            f.write(str(position)+\"\\n\")\n",
    "    f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def five_end_mapping_opposite_stallreporter(samgen,fasta):\n",
    "    from BCBio import GFF\n",
    "    from Bio import Seq\n",
    "    outputdata1={}\n",
    "    outputdata2={}\n",
    "    #######################################################\n",
    "    with open(fasta,'r') as f:\n",
    "        sequence = ''\n",
    "        for line in f:\n",
    "            if line[0] == '>':\n",
    "                continue\n",
    "            sequence = sequence+line\n",
    "    #######################################################    \n",
    "    outputdata1['plus']=[0 for x in range(len(sequence))]\n",
    "    outputdata2['minus']=[0 for x in range(len(sequence))]\n",
    "    \n",
    "    for read in samgen:\n",
    "        if read[0][0] == '@':   \n",
    "            continue\n",
    "        if read[1] == '4':      \n",
    "            continue            \n",
    "        chrom = read[2]             \n",
    "        readid = read[0]            \n",
    "        startp = int(read[3]) - 1   \n",
    "        seq = Seq.Seq(read[9])      \n",
    "        length = len(seq)           \n",
    "        if (read[1] == '16'):\n",
    "            start = startp + length - 1\n",
    "            outputdata1['plus'][start] += 1 ##### \n",
    "        if (read[1] == '0'):\n",
    "            start = startp\n",
    "            outputdata2['minus'][start] += 1 ##### \n",
    "    return [outputdata1,outputdata2] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "filepath = ''\n",
    "fastq_path = filepath+'fastq_3RACE/'\n",
    "fasta_path = filepath+'StallReporters/'\n",
    "\n",
    "files = ['ks'+str(x) for x in range(246,250)]+['ks'+str(x) for x in range(254,258)]\n",
    "\n",
    "# cutadapt\n",
    "cutadapt_input_extension = '.fastq'\n",
    "cutadapt_output_extension = '-ctadpt.fastq'\n",
    "cutadapt_adapter = 'TCCTTGGTGCCCGAGTGNNNNNN'\n",
    "cutadapt_option = '--minimum-length 11 --discard-untrimmed'\n",
    "cutadapt_output_extension_3end = '-ctadpt_ctadpt3.fastq'\n",
    "cutadapt_adapter_3end = 'AGATCGGAAGAGCACACGTC'\n",
    "cutadapt_option_3end = '--minimum-length 11'\n",
    "\n",
    "# bowtie\n",
    "inpath = fastq_path\n",
    "outpath = filepath\n",
    "bowtie_input_extension = cutadapt_output_extension_3end\n",
    "\n",
    "# mapping\n",
    "mapping = '_5map_opposite'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ks246\n",
      "ks247\n",
      "ks248\n",
      "ks249\n",
      "ks254\n",
      "ks255\n",
      "ks256\n",
      "ks257\n"
     ]
    }
   ],
   "source": [
    "log_file = filepath+'log_3RACE_cutadapt.txt'\n",
    "###\n",
    "logs = open(log_file, 'w')\n",
    "logs.close()\n",
    "###\n",
    "\n",
    "for f in files:\n",
    "    print f\n",
    "    ###\n",
    "    cutadapt_inputfile = fastq_path+f+cutadapt_input_extension\n",
    "    cutadapt_outputfile = fastq_path+f+cutadapt_output_extension\n",
    "\n",
    "    command = 'cutadapt -g ^%s %s -o %s %s >> %s' % (\n",
    "        cutadapt_adapter,\n",
    "        cutadapt_option, \n",
    "        cutadapt_outputfile,\n",
    "        cutadapt_inputfile,\n",
    "        log_file\n",
    "        )\n",
    "    \n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n\\n'+f+'\\n')\n",
    "    logs.close()\n",
    "    \n",
    "    os.system(command)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ks246\n",
      "ks247\n",
      "ks248\n",
      "ks249\n",
      "ks254\n",
      "ks255\n",
      "ks256\n",
      "ks257\n"
     ]
    }
   ],
   "source": [
    "log_file = filepath+'log_3RACE_cutadapt_3end.txt'\n",
    "###\n",
    "logs = open(log_file, 'w')\n",
    "logs.close()\n",
    "###\n",
    "\n",
    "for f in files:\n",
    "    print f\n",
    "    ###\n",
    "    cutadapt_inputfile = fastq_path+f+cutadapt_output_extension\n",
    "    cutadapt_outputfile = fastq_path+f+cutadapt_output_extension_3end\n",
    "\n",
    "    command = 'cutadapt -a %s %s -o %s %s >> %s' % (\n",
    "        cutadapt_adapter_3end,\n",
    "        cutadapt_option_3end, \n",
    "        cutadapt_outputfile,\n",
    "        cutadapt_inputfile,\n",
    "        log_file\n",
    "        )\n",
    "    \n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n\\n'+f+'\\n')\n",
    "    logs.close()\n",
    "    \n",
    "    os.system(command)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ks246\n",
      "ks247\n",
      "ks248\n",
      "ks249\n"
     ]
    }
   ],
   "source": [
    "######################################################\n",
    "\n",
    "files_reporter = ['ks'+str(x) for x in range(246,250)]\n",
    "stall_reporter = 'secM'\n",
    "\n",
    "######################################################\n",
    "\n",
    "###\n",
    "path_stallreporter = outpath + 'alignments/StallReporter_'+stall_reporter+'/'\n",
    "if not os.path.exists(outpath + 'alignments/'):\n",
    "    os.makedirs(outpath + 'alignments/')\n",
    "if not os.path.exists(path_stallreporter):\n",
    "    os.makedirs(path_stallreporter)\n",
    "###   \n",
    "log_file = filepath+'log_3RACE_Bowtie_StallReporter_'+stall_reporter+'.txt'\n",
    "logs = open(log_file, 'w')\n",
    "logs.close()\n",
    "###\n",
    "\n",
    "for fname in files_reporter:\n",
    "    print fname\n",
    "\n",
    "    bowtie_command = 'bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 '\\\n",
    "        '--un %s%s --max %s%s --al %s%s %s %s%s %s%s 1>> %s 2>> %s'\n",
    " \n",
    "    bowtie_stallreporter = bowtie_command %(\n",
    "        path_stallreporter,\n",
    "        fname+'_nomatch.fastq',\n",
    "        path_stallreporter,\n",
    "        fname+'_multi.fastq',\n",
    "        path_stallreporter,\n",
    "        fname+'_match.fastq',\n",
    "        fasta_path+'StallReporter_'+stall_reporter,\n",
    "        inpath,\n",
    "        fname + bowtie_input_extension,\n",
    "        path_stallreporter,\n",
    "        fname + '_StallReporter_'+stall_reporter+'_match.SAM',\n",
    "        log_file,\n",
    "        log_file\n",
    "        )\n",
    "    ###\n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n\\n'+fname+'\\n'+bowtie_stallreporter+'\\n')\n",
    "    logs.close()\n",
    "    ###\n",
    "    os.system(bowtie_stallreporter)           "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fasta = fasta_path+'StallReporter_'+stall_reporter+'.fasta'\n",
    "\n",
    "out_path=filepath+'density_StallReporter/'\n",
    "if not os.path.exists(out_path):\n",
    "    os.makedirs(out_path)\n",
    "\n",
    "for f in files_reporter:\n",
    "    samgenmain_filename=filepath+'alignments/StallReporter_'+stall_reporter+'/'+f+'_StallReporter_'+stall_reporter+'_match.SAM'\n",
    "    f_samgenmain=open(samgenmain_filename)\n",
    "    samgenmain=csv.reader(f_samgenmain,delimiter='\\t')\n",
    "\n",
    "    three_mapped=five_end_mapping_opposite_stallreporter(samgenmain,fasta)\n",
    "    ###\n",
    "    pickle_readcounts = out_path+f+stall_reporter+mapping+'_readcounts.pickle'\n",
    "    with open(pickle_readcounts, 'wb') as readcounts_f:\n",
    "        pickle.dump(three_mapped,readcounts_f)\n",
    "    ###\n",
    "    three_mapped_rpm = deepcopy(three_mapped)\n",
    "    readcounts_to_rpm(three_mapped_rpm[0])\n",
    "    #readcounts_to_rpm(three_mapped_rpm[1])\n",
    "    pickle_rpm = out_path+f+stall_reporter+mapping+'_rpm.pickle'\n",
    "    with open(pickle_rpm, 'wb') as rpm_f:\n",
    "        pickle.dump(three_mapped_rpm,rpm_f)\n",
    "\n",
    "    density_path = out_path+f+stall_reporter+mapping\n",
    "    writecountsf(three_mapped_rpm[0],density_path+\"_plus_\")\n",
    "    countstowig(three_mapped_rpm[0],density_path+\"_plus\")\n",
    "    writecountsf(three_mapped_rpm[1],density_path+\"_minus_\")\n",
    "    countstowig(three_mapped_rpm[1],density_path+\"_minus\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ks254\n",
      "ks255\n",
      "ks256\n",
      "ks257\n"
     ]
    }
   ],
   "source": [
    "######################################################\n",
    "\n",
    "files_reporter = ['ks'+str(x) for x in range(254,258)]\n",
    "stall_reporter = 'EPstop'\n",
    "\n",
    "######################################################\n",
    "\n",
    "###\n",
    "path_stallreporter = outpath + 'alignments/StallReporter_'+stall_reporter+'/'\n",
    "if not os.path.exists(outpath + 'alignments/'):\n",
    "    os.makedirs(outpath + 'alignments/')\n",
    "if not os.path.exists(path_stallreporter):\n",
    "    os.makedirs(path_stallreporter)\n",
    "###   \n",
    "log_file = filepath+'log_3RACE_Bowtie_StallReporter_'+stall_reporter+'.txt'\n",
    "logs = open(log_file, 'w')\n",
    "logs.close()\n",
    "###\n",
    "\n",
    "for fname in files_reporter:\n",
    "    print fname\n",
    "\n",
    "    bowtie_command = 'bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 '\\\n",
    "        '--un %s%s --max %s%s --al %s%s %s %s%s %s%s 1>> %s 2>> %s'\n",
    " \n",
    "    bowtie_stallreporter = bowtie_command %(\n",
    "        path_stallreporter,\n",
    "        fname+'_nomatch.fastq',\n",
    "        path_stallreporter,\n",
    "        fname+'_multi.fastq',\n",
    "        path_stallreporter,\n",
    "        fname+'_match.fastq',\n",
    "        fasta_path+'StallReporter_'+stall_reporter,\n",
    "        inpath,\n",
    "        fname + bowtie_input_extension,\n",
    "        path_stallreporter,\n",
    "        fname + '_StallReporter_'+stall_reporter+'_match.SAM',\n",
    "        log_file,\n",
    "        log_file\n",
    "        )\n",
    "    ###\n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n\\n'+fname+'\\n'+bowtie_stallreporter+'\\n')\n",
    "    logs.close()\n",
    "    ###\n",
    "    os.system(bowtie_stallreporter)           "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fasta = fasta_path+'StallReporter_'+stall_reporter+'.fasta'\n",
    "\n",
    "out_path=filepath+'density_StallReporter/'\n",
    "if not os.path.exists(out_path):\n",
    "    os.makedirs(out_path)\n",
    "\n",
    "for f in files_reporter:\n",
    "    samgenmain_filename=filepath+'alignments/StallReporter_'+stall_reporter+'/'+f+'_StallReporter_'+stall_reporter+'_match.SAM'\n",
    "    f_samgenmain=open(samgenmain_filename)\n",
    "    samgenmain=csv.reader(f_samgenmain,delimiter='\\t')\n",
    "\n",
    "    three_mapped=five_end_mapping_opposite_stallreporter(samgenmain,fasta)\n",
    "    ###\n",
    "    pickle_readcounts = out_path+f+stall_reporter+mapping+'_readcounts.pickle'\n",
    "    with open(pickle_readcounts, 'wb') as readcounts_f:\n",
    "        pickle.dump(three_mapped,readcounts_f)\n",
    "    ###\n",
    "    three_mapped_rpm = deepcopy(three_mapped)\n",
    "    readcounts_to_rpm(three_mapped_rpm[0])\n",
    "    #readcounts_to_rpm(three_mapped_rpm[1])\n",
    "    pickle_rpm = out_path+f+stall_reporter+mapping+'_rpm.pickle'\n",
    "    with open(pickle_rpm, 'wb') as rpm_f:\n",
    "        pickle.dump(three_mapped_rpm,rpm_f)\n",
    "\n",
    "    density_path = out_path+f+stall_reporter+mapping\n",
    "    writecountsf(three_mapped_rpm[0],density_path+\"_plus_\")\n",
    "    countstowig(three_mapped_rpm[0],density_path+\"_plus\")\n",
    "    writecountsf(three_mapped_rpm[1],density_path+\"_minus_\")\n",
    "    countstowig(three_mapped_rpm[1],density_path+\"_minus\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
