{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "######################################################\n",
    "import csv\n",
    "import pickle\n",
    "from copy import deepcopy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def readcounts_to_rpm(readcounts):\n",
    "    for chrom in readcounts.keys():\n",
    "        totalread = sum(readcounts[chrom])\n",
    "        for position in range(len(readcounts[chrom])):\n",
    "            readcounts[chrom][position]/=float(totalread)\n",
    "            readcounts[chrom][position]*=1E6\n",
    "            \n",
    "def writecountsf(rpm,filepath): \n",
    "    import struct\n",
    "    f2=open(filepath+\"keys\",\"w\")\n",
    "    for chrom in rpm.keys():\n",
    "        f=open(filepath+\"genome\",\"wb\")\n",
    "        for position in rpm[chrom]:\n",
    "            f.write(struct.pack(\"f\",position))\n",
    "        f.close()\n",
    "        f2.write(chrom+\"\\n\")\n",
    "    f2.close()\n",
    "\n",
    "def countstowig(rpm,filestring):\n",
    "    import random\n",
    "    f=open(filestring+\".wig\",\"w\")\n",
    "    filestring=filestring.partition(\"_\")[0][-3:]\n",
    "\n",
    "    random.seed(filestring)\n",
    "    c1=random.randint(0,255)\n",
    "    random.seed(filestring+\"xxx\")\n",
    "    c2=random.randint(0,255)\n",
    "    random.seed(filestring+\"000\")\n",
    "    c3=random.randint(0,255)\n",
    "\n",
    "    f.write(\"track name=tracklabel viewLimits=-5:5 color=\"+str(c1)+','+str(c2)+','+str(c3)+\"\\n\")\n",
    "    for chrom in rpm.keys():\n",
    "        if chrom[0:3]=='chr':\n",
    "            f.write(\"fixedStep  chrom=\"+chrom+\"  start=1  step=1\\n\")\n",
    "        else:\n",
    "            f.write(\"fixedStep  chrom=\\\"\"+chrom+\"\\\"  start=1  step=1\\n\")\n",
    "\n",
    "\n",
    "        for position in rpm[chrom]:\n",
    "            f.write(str(position)+\"\\n\")\n",
    "    f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def five_end_mapping_stallreporter(samgen,fasta):\n",
    "    from BCBio import GFF\n",
    "    from Bio import Seq\n",
    "    outputdata1={}\n",
    "    outputdata2={}\n",
    "    #######################################################\n",
    "    with open(fasta,'r') as f:\n",
    "        sequence = ''\n",
    "        for line in f:\n",
    "            if line[0] == '>':\n",
    "                continue\n",
    "            sequence = sequence+line\n",
    "    #######################################################    \n",
    "    outputdata1['plus']=[0 for x in range(len(sequence))]\n",
    "    outputdata2['minus']=[0 for x in range(len(sequence))]\n",
    "    \n",
    "    for read in samgen:\n",
    "        if read[0][0] == '@':   \n",
    "            continue\n",
    "        if read[1] == '4':      \n",
    "            continue            \n",
    "        chrom = read[2]             \n",
    "        readid = read[0]            \n",
    "        startp = int(read[3]) - 1    \n",
    "        seq = Seq.Seq(read[9])      \n",
    "        length = len(seq)           \n",
    "        if (read[1] == '16'):\n",
    "            start = startp + length - 1\n",
    "            outputdata2['minus'][start] += 1 ##### \n",
    "        if (read[1] == '0'):\n",
    "            start = startp\n",
    "            outputdata1['plus'][start] += 1 ##### \n",
    "    return [outputdata1,outputdata2] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "filepath = ''\n",
    "fastq_path = filepath+'fastq_5RACE/'\n",
    "fasta_path = filepath+'StallReporters/'\n",
    "\n",
    "files = ['ks'+str(x) for x in range(222,226)]+['ks'+str(x) for x in range(230,234)]\n",
    "\n",
    "# cutadapt\n",
    "cutadapt_input_extension = '.fastq'\n",
    "cutadapt_output_extension = '-ctadpt.fastq'\n",
    "cutadapt_adapter = 'TGCCCGAGTG'\n",
    "cutadapt_option = '--minimum-length 11 --discard-untrimmed'\n",
    "cutadapt_output_extension_3end = '-ctadpt_ctadpt3.fastq'\n",
    "cutadapt_adapter_3end = 'GCGGTCGAGTTCTGGACCGA'\n",
    "cutadapt_option_3end = '--minimum-length 11'\n",
    "\n",
    "# bowtie\n",
    "inpath = fastq_path\n",
    "outpath = filepath\n",
    "bowtie_input_extension = cutadapt_output_extension_3end\n",
    "\n",
    "# mapping\n",
    "mapping = '_5map'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ks222\n",
      "ks223\n",
      "ks224\n",
      "ks225\n",
      "ks230\n",
      "ks231\n",
      "ks232\n",
      "ks233\n",
      "\n",
      "\n",
      "\n",
      "ks222\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -g ^TGCCCGAGTG --minimum-length 11 --discard-untrimmed -o fastq_5RACE/ks222-ctadpt.fastq fastq_5RACE/ks222.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.31 s (8 us/read; 7.26 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  37,788\n",
      "Reads with adapters:                    33,175 (87.8%)\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        33,175 (87.8%)\n",
      "\n",
      "Total basepairs processed:     5,705,988 bp\n",
      "Total written (filtered):      4,678,812 bp (82.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGCCCGAGTG; Type: anchored 5'; Length: 10; Trimmed: 33175 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1490\t0.1\t0\t0 1490\n",
      "10\t31332\t0.0\t1\t30779 553\n",
      "11\t353\t0.0\t1\t0 353\n",
      "\n",
      "\n",
      "\n",
      "ks223\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -g ^TGCCCGAGTG --minimum-length 11 --discard-untrimmed -o fastq_5RACE/ks223-ctadpt.fastq fastq_5RACE/ks223.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.27 s (8 us/read; 7.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  32,308\n",
      "Reads with adapters:                    28,707 (88.9%)\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        28,707 (88.9%)\n",
      "\n",
      "Total basepairs processed:     4,878,508 bp\n",
      "Total written (filtered):      4,048,611 bp (83.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGCCCGAGTG; Type: anchored 5'; Length: 10; Trimmed: 28707 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1219\t0.1\t0\t0 1219\n",
      "10\t27193\t0.0\t1\t26655 538\n",
      "11\t295\t0.0\t1\t0 295\n",
      "\n",
      "\n",
      "\n",
      "ks224\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -g ^TGCCCGAGTG --minimum-length 11 --discard-untrimmed -o fastq_5RACE/ks224-ctadpt.fastq fastq_5RACE/ks224.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.35 s (8 us/read; 7.77 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  45,146\n",
      "Reads with adapters:                    38,463 (85.2%)\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        38,463 (85.2%)\n",
      "\n",
      "Total basepairs processed:     6,817,046 bp\n",
      "Total written (filtered):      5,424,453 bp (79.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGCCCGAGTG; Type: anchored 5'; Length: 10; Trimmed: 38463 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1662\t0.2\t0\t0 1662\n",
      "10\t36309\t0.0\t1\t35619 690\n",
      "11\t492\t0.0\t1\t0 492\n",
      "\n",
      "\n",
      "\n",
      "ks225\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -g ^TGCCCGAGTG --minimum-length 11 --discard-untrimmed -o fastq_5RACE/ks225-ctadpt.fastq fastq_5RACE/ks225.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.40 s (8 us/read; 7.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  47,845\n",
      "Reads with adapters:                    41,469 (86.7%)\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        41,469 (86.7%)\n",
      "\n",
      "Total basepairs processed:     7,224,595 bp\n",
      "Total written (filtered):      5,848,302 bp (80.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGCCCGAGTG; Type: anchored 5'; Length: 10; Trimmed: 41469 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1699\t0.2\t0\t0 1699\n",
      "10\t39244\t0.0\t1\t38501 743\n",
      "11\t526\t0.0\t1\t0 526\n",
      "\n",
      "\n",
      "\n",
      "ks230\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -g ^TGCCCGAGTG --minimum-length 11 --discard-untrimmed -o fastq_5RACE/ks230-ctadpt.fastq fastq_5RACE/ks230.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.32 s (8 us/read; 7.26 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  38,965\n",
      "Reads with adapters:                    34,544 (88.7%)\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        34,544 (88.7%)\n",
      "\n",
      "Total basepairs processed:     5,883,715 bp\n",
      "Total written (filtered):      4,873,104 bp (82.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGCCCGAGTG; Type: anchored 5'; Length: 10; Trimmed: 34544 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2825\t0.1\t0\t0 2825\n",
      "10\t31294\t0.0\t1\t30713 581\n",
      "11\t425\t0.0\t1\t0 425\n",
      "\n",
      "\n",
      "\n",
      "ks231\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -g ^TGCCCGAGTG --minimum-length 11 --discard-untrimmed -o fastq_5RACE/ks231-ctadpt.fastq fastq_5RACE/ks231.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.35 s (9 us/read; 7.00 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  40,702\n",
      "Reads with adapters:                    37,073 (91.1%)\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        37,073 (91.1%)\n",
      "\n",
      "Total basepairs processed:     6,146,002 bp\n",
      "Total written (filtered):      5,229,639 bp (85.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGCCCGAGTG; Type: anchored 5'; Length: 10; Trimmed: 37073 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2829\t0.2\t0\t0 2829\n",
      "10\t33761\t0.0\t1\t33152 609\n",
      "11\t483\t0.0\t1\t0 483\n",
      "\n",
      "\n",
      "\n",
      "ks232\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -g ^TGCCCGAGTG --minimum-length 11 --discard-untrimmed -o fastq_5RACE/ks232-ctadpt.fastq fastq_5RACE/ks232.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.35 s (8 us/read; 7.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  41,742\n",
      "Reads with adapters:                    36,091 (86.5%)\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        36,091 (86.5%)\n",
      "\n",
      "Total basepairs processed:     6,303,042 bp\n",
      "Total written (filtered):      5,091,414 bp (80.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGCCCGAGTG; Type: anchored 5'; Length: 10; Trimmed: 36091 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2998\t0.2\t0\t0 2998\n",
      "10\t32678\t0.0\t1\t32090 588\n",
      "11\t415\t0.0\t1\t0 415\n",
      "\n",
      "\n",
      "\n",
      "ks233\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -g ^TGCCCGAGTG --minimum-length 11 --discard-untrimmed -o fastq_5RACE/ks233-ctadpt.fastq fastq_5RACE/ks233.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.39 s (8 us/read; 7.26 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  46,766\n",
      "Reads with adapters:                    39,853 (85.2%)\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        39,853 (85.2%)\n",
      "\n",
      "Total basepairs processed:     7,061,666 bp\n",
      "Total written (filtered):      5,622,070 bp (79.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGCCCGAGTG; Type: anchored 5'; Length: 10; Trimmed: 39853 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10 bp: 1\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t3280\t0.2\t0\t0 3280\n",
      "10\t36090\t0.0\t1\t35446 644\n",
      "11\t483\t0.0\t1\t0 483\n",
      "\n"
     ]
    }
   ],
   "source": [
    "log_file = filepath+'log_5RACE_cutadapt.txt'\n",
    "###\n",
    "logs = open(log_file, 'w')\n",
    "logs.close()\n",
    "###\n",
    "\n",
    "for f in files:\n",
    "    print f\n",
    "    ###\n",
    "    cutadapt_inputfile = fastq_path+f+cutadapt_input_extension\n",
    "    cutadapt_outputfile = fastq_path+f+cutadapt_output_extension\n",
    "\n",
    "    command = 'cutadapt -g ^%s %s -o %s %s >> %s' % (\n",
    "        cutadapt_adapter,\n",
    "        cutadapt_option, \n",
    "        cutadapt_outputfile,\n",
    "        cutadapt_inputfile,\n",
    "        log_file\n",
    "        )\n",
    "    \n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n\\n'+f+'\\n')\n",
    "    logs.close()\n",
    "    \n",
    "    os.system(command)\n",
    "    ###\n",
    "\n",
    "print \n",
    "for line in open(log_file, 'r'):\n",
    "    print line[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ks222\n",
      "ks223\n",
      "ks224\n",
      "ks225\n",
      "ks230\n",
      "ks231\n",
      "ks232\n",
      "ks233\n",
      "\n",
      "\n",
      "\n",
      "ks222\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -a GCGGTCGAGTTCTGGACCGA --minimum-length 11 -o fastq_5RACE/ks222-ctadpt_ctadpt3.fastq fastq_5RACE/ks222-ctadpt.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.31 s (9 us/read; 6.39 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  33,175\n",
      "Reads with adapters:                    29,370 (88.5%)\n",
      "Reads that were too short:               3,876 (11.7%)\n",
      "Reads written (passing filters):        29,299 (88.3%)\n",
      "\n",
      "Total basepairs processed:     4,678,812 bp\n",
      "Total written (filtered):      2,223,331 bp (47.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCGGTCGAGTTCTGGACCGA; Type: regular 3'; Length: 20; Trimmed: 29370 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 98.6%\n",
      "  C: 0.2%\n",
      "  G: 0.1%\n",
      "  T: 0.1%\n",
      "  none/other: 1.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"A\" extremely often.\n",
      "    The provided adapter sequence may be incomplete.\n",
      "    To fix the problem, add \"A\" to the beginning of the adapter sequence.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t86\t518.4\t0\t86\n",
      "4\t14\t129.6\t0\t14\n",
      "5\t15\t32.4\t0\t15\n",
      "6\t19\t8.1\t0\t19\n",
      "7\t30\t2.0\t0\t30\n",
      "8\t19\t0.5\t0\t19\n",
      "9\t48\t0.1\t0\t48\n",
      "10\t36\t0.0\t1\t36\n",
      "11\t73\t0.0\t1\t71 2\n",
      "12\t19\t0.0\t1\t18 1\n",
      "13\t80\t0.0\t1\t80\n",
      "14\t8\t0.0\t1\t8\n",
      "15\t31\t0.0\t1\t30 1\n",
      "16\t55\t0.0\t1\t54 1\n",
      "17\t14\t0.0\t1\t12 2\n",
      "18\t21\t0.0\t1\t20 1\n",
      "19\t16\t0.0\t1\t16\n",
      "20\t14\t0.0\t2\t13 1\n",
      "21\t10\t0.0\t2\t10\n",
      "22\t11\t0.0\t2\t11\n",
      "23\t14\t0.0\t2\t13 1\n",
      "24\t6\t0.0\t2\t6\n",
      "25\t8\t0.0\t2\t7 1\n",
      "26\t10\t0.0\t2\t9 1\n",
      "27\t18\t0.0\t2\t18\n",
      "28\t15\t0.0\t2\t14 1\n",
      "29\t28\t0.0\t2\t28\n",
      "30\t31\t0.0\t2\t30 0 1\n",
      "31\t35\t0.0\t2\t33 2\n",
      "32\t34\t0.0\t2\t33 0 1\n",
      "33\t27\t0.0\t2\t27\n",
      "34\t18\t0.0\t2\t18\n",
      "35\t17\t0.0\t2\t17\n",
      "36\t63\t0.0\t2\t62 1\n",
      "37\t48\t0.0\t2\t45 3\n",
      "38\t12\t0.0\t2\t11 1\n",
      "39\t269\t0.0\t2\t266 2 1\n",
      "40\t87\t0.0\t2\t84 3\n",
      "41\t37\t0.0\t2\t35 1 1\n",
      "42\t62\t0.0\t2\t60 2\n",
      "43\t205\t0.0\t2\t198 6 1\n",
      "44\t16\t0.0\t2\t16\n",
      "45\t17\t0.0\t2\t15 0 2\n",
      "46\t11\t0.0\t2\t11\n",
      "47\t21\t0.0\t2\t19 1 1\n",
      "48\t60\t0.0\t2\t60\n",
      "49\t61\t0.0\t2\t56 2 3\n",
      "50\t63\t0.0\t2\t62 1\n",
      "51\t36\t0.0\t2\t36\n",
      "52\t29\t0.0\t2\t29\n",
      "53\t26\t0.0\t2\t25 1\n",
      "54\t17\t0.0\t2\t16 1\n",
      "55\t38\t0.0\t2\t36 2\n",
      "56\t27\t0.0\t2\t27\n",
      "57\t33\t0.0\t2\t33\n",
      "58\t33\t0.0\t2\t31 2\n",
      "59\t68\t0.0\t2\t62 6\n",
      "60\t106\t0.0\t2\t99 5 2\n",
      "61\t152\t0.0\t2\t146 6\n",
      "62\t86\t0.0\t2\t84 2\n",
      "63\t58\t0.0\t2\t55 3\n",
      "64\t79\t0.0\t2\t77 2\n",
      "65\t210\t0.0\t2\t207 3\n",
      "66\t6942\t0.0\t2\t6786 139 17\n",
      "67\t7286\t0.0\t2\t7105 167 14\n",
      "68\t117\t0.0\t2\t116 1\n",
      "69\t84\t0.0\t2\t84\n",
      "70\t28\t0.0\t2\t27 1\n",
      "71\t35\t0.0\t2\t35\n",
      "72\t30\t0.0\t2\t29 1\n",
      "73\t32\t0.0\t2\t31 1\n",
      "74\t39\t0.0\t2\t36 3\n",
      "75\t30\t0.0\t2\t30\n",
      "76\t27\t0.0\t2\t27\n",
      "77\t13\t0.0\t2\t13\n",
      "78\t43\t0.0\t2\t42 1\n",
      "79\t83\t0.0\t2\t78 4 1\n",
      "80\t172\t0.0\t2\t167 4 1\n",
      "81\t256\t0.0\t2\t248 8\n",
      "82\t266\t0.0\t2\t260 6\n",
      "83\t257\t0.0\t2\t251 5 1\n",
      "84\t172\t0.0\t2\t172\n",
      "85\t187\t0.0\t2\t181 5 1\n",
      "86\t515\t0.0\t2\t506 7 2\n",
      "87\t159\t0.0\t2\t153 5 1\n",
      "88\t166\t0.0\t2\t164 2\n",
      "89\t142\t0.0\t2\t140 2\n",
      "90\t85\t0.0\t2\t83 2\n",
      "91\t59\t0.0\t2\t59\n",
      "92\t135\t0.0\t2\t133 2\n",
      "93\t168\t0.0\t2\t165 3\n",
      "94\t193\t0.0\t2\t189 3 1\n",
      "95\t337\t0.0\t2\t324 13\n",
      "96\t689\t0.0\t2\t670 18 1\n",
      "97\t287\t0.0\t2\t282 3 2\n",
      "98\t143\t0.0\t2\t140 3\n",
      "99\t102\t0.0\t2\t101 1\n",
      "100\t101\t0.0\t2\t97 4\n",
      "101\t112\t0.0\t2\t110 2\n",
      "102\t127\t0.0\t2\t126 1\n",
      "103\t96\t0.0\t2\t95 1\n",
      "104\t26\t0.0\t2\t25 1\n",
      "105\t7\t0.0\t2\t7\n",
      "106\t7\t0.0\t2\t7\n",
      "107\t13\t0.0\t2\t12 1\n",
      "108\t22\t0.0\t2\t21 1\n",
      "109\t24\t0.0\t2\t24\n",
      "110\t31\t0.0\t2\t31\n",
      "111\t20\t0.0\t2\t20\n",
      "112\t24\t0.0\t2\t24\n",
      "113\t28\t0.0\t2\t27 1\n",
      "114\t28\t0.0\t2\t28\n",
      "115\t49\t0.0\t2\t48 0 1\n",
      "116\t55\t0.0\t2\t49 5 1\n",
      "117\t67\t0.0\t2\t64 3\n",
      "118\t154\t0.0\t2\t149 5\n",
      "119\t106\t0.0\t2\t101 5\n",
      "120\t92\t0.0\t2\t89 2 1\n",
      "121\t58\t0.0\t2\t57 1\n",
      "122\t121\t0.0\t2\t115 5 1\n",
      "123\t83\t0.0\t2\t82 1\n",
      "124\t97\t0.0\t2\t93 4\n",
      "125\t160\t0.0\t2\t152 8\n",
      "126\t162\t0.0\t2\t156 4 2\n",
      "127\t142\t0.0\t2\t138 4\n",
      "128\t237\t0.0\t2\t229 8\n",
      "129\t515\t0.0\t2\t497 17 1\n",
      "130\t552\t0.0\t2\t532 19 1\n",
      "131\t644\t0.0\t2\t633 11\n",
      "132\t641\t0.0\t2\t628 12 1\n",
      "133\t367\t0.0\t2\t358 9\n",
      "134\t779\t0.0\t2\t767 12\n",
      "135\t170\t0.0\t2\t161 7 2\n",
      "136\t133\t0.0\t2\t131 2\n",
      "137\t265\t0.0\t2\t259 6\n",
      "138\t142\t0.0\t2\t140 2\n",
      "139\t180\t0.0\t2\t172 6 2\n",
      "140\t249\t0.0\t2\t238 9 2\n",
      "141\t292\t0.0\t2\t56 119 117\n",
      "142\t3\t0.0\t2\t1 1 1\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "\n",
      "ks223\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -a GCGGTCGAGTTCTGGACCGA --minimum-length 11 -o fastq_5RACE/ks223-ctadpt_ctadpt3.fastq fastq_5RACE/ks223-ctadpt.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.26 s (9 us/read; 6.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  28,707\n",
      "Reads with adapters:                    24,825 (86.5%)\n",
      "Reads that were too short:               3,299 (11.5%)\n",
      "Reads written (passing filters):        25,408 (88.5%)\n",
      "\n",
      "Total basepairs processed:     4,048,611 bp\n",
      "Total written (filtered):      1,990,162 bp (49.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCGGTCGAGTTCTGGACCGA; Type: regular 3'; Length: 20; Trimmed: 24825 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 98.0%\n",
      "  C: 0.3%\n",
      "  G: 0.1%\n",
      "  T: 0.1%\n",
      "  none/other: 1.5%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"A\" extremely often.\n",
      "    The provided adapter sequence may be incomplete.\n",
      "    To fix the problem, add \"A\" to the beginning of the adapter sequence.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t92\t448.5\t0\t92\n",
      "4\t14\t112.1\t0\t14\n",
      "5\t23\t28.0\t0\t23\n",
      "6\t41\t7.0\t0\t41\n",
      "7\t29\t1.8\t0\t29\n",
      "8\t34\t0.4\t0\t34\n",
      "9\t113\t0.1\t0\t113\n",
      "10\t48\t0.0\t1\t47 1\n",
      "11\t155\t0.0\t1\t155\n",
      "12\t37\t0.0\t1\t35 2\n",
      "13\t84\t0.0\t1\t83 1\n",
      "14\t14\t0.0\t1\t13 1\n",
      "15\t53\t0.0\t1\t53\n",
      "16\t71\t0.0\t1\t70 1\n",
      "17\t12\t0.0\t1\t12\n",
      "18\t25\t0.0\t1\t24 1\n",
      "19\t18\t0.0\t1\t18\n",
      "20\t18\t0.0\t2\t15 2 1\n",
      "21\t17\t0.0\t2\t16 1\n",
      "22\t11\t0.0\t2\t11\n",
      "23\t9\t0.0\t2\t8 1\n",
      "24\t14\t0.0\t2\t13 1\n",
      "25\t9\t0.0\t2\t9\n",
      "26\t8\t0.0\t2\t7 1\n",
      "27\t12\t0.0\t2\t11 0 1\n",
      "28\t15\t0.0\t2\t15\n",
      "29\t49\t0.0\t2\t49\n",
      "30\t53\t0.0\t2\t52 1\n",
      "31\t53\t0.0\t2\t51 1 1\n",
      "32\t52\t0.0\t2\t51 0 1\n",
      "33\t51\t0.0\t2\t49 2\n",
      "34\t37\t0.0\t2\t37\n",
      "35\t17\t0.0\t2\t16 1\n",
      "36\t107\t0.0\t2\t107\n",
      "37\t36\t0.0\t2\t35 0 1\n",
      "38\t15\t0.0\t2\t14 0 1\n",
      "39\t295\t0.0\t2\t287 8\n",
      "40\t102\t0.0\t2\t102\n",
      "41\t38\t0.0\t2\t37 1\n",
      "42\t90\t0.0\t2\t89 1\n",
      "43\t322\t0.0\t2\t314 7 1\n",
      "44\t10\t0.0\t2\t10\n",
      "45\t26\t0.0\t2\t25 1\n",
      "46\t23\t0.0\t2\t23\n",
      "47\t26\t0.0\t2\t24 0 2\n",
      "48\t67\t0.0\t2\t64 3\n",
      "49\t88\t0.0\t2\t86 2\n",
      "50\t75\t0.0\t2\t70 3 2\n",
      "51\t26\t0.0\t2\t25 1\n",
      "52\t27\t0.0\t2\t25 2\n",
      "53\t16\t0.0\t2\t16\n",
      "54\t25\t0.0\t2\t25\n",
      "55\t34\t0.0\t2\t32 2\n",
      "56\t25\t0.0\t2\t25\n",
      "57\t32\t0.0\t2\t31 1\n",
      "58\t37\t0.0\t2\t36 1\n",
      "59\t62\t0.0\t2\t59 2 1\n",
      "60\t124\t0.0\t2\t121 3\n",
      "61\t158\t0.0\t2\t153 3 2\n",
      "62\t98\t0.0\t2\t95 3\n",
      "63\t46\t0.0\t2\t44 1 1\n",
      "64\t53\t0.0\t2\t51 1 1\n",
      "65\t176\t0.0\t2\t173 3\n",
      "66\t6961\t0.0\t2\t6808 141 12\n",
      "67\t2903\t0.0\t2\t2817 78 8\n",
      "68\t103\t0.0\t2\t98 4 1\n",
      "69\t79\t0.0\t2\t76 3\n",
      "70\t30\t0.0\t2\t28 2\n",
      "71\t40\t0.0\t2\t39 1\n",
      "72\t19\t0.0\t2\t16 3\n",
      "73\t21\t0.0\t2\t21\n",
      "74\t41\t0.0\t2\t41\n",
      "75\t34\t0.0\t2\t34\n",
      "76\t20\t0.0\t2\t19 1\n",
      "77\t22\t0.0\t2\t20 2\n",
      "78\t37\t0.0\t2\t34 2 1\n",
      "79\t83\t0.0\t2\t79 2 2\n",
      "80\t167\t0.0\t2\t163 4\n",
      "81\t274\t0.0\t2\t269 3 2\n",
      "82\t240\t0.0\t2\t232 8\n",
      "83\t264\t0.0\t2\t259 4 1\n",
      "84\t211\t0.0\t2\t207 4\n",
      "85\t220\t0.0\t2\t214 5 1\n",
      "86\t699\t0.0\t2\t688 9 2\n",
      "87\t219\t0.0\t2\t215 4\n",
      "88\t196\t0.0\t2\t189 5 2\n",
      "89\t134\t0.0\t2\t129 5\n",
      "90\t76\t0.0\t2\t74 2\n",
      "91\t62\t0.0\t2\t61 1\n",
      "92\t93\t0.0\t2\t90 3\n",
      "93\t147\t0.0\t2\t142 5\n",
      "94\t173\t0.0\t2\t170 3\n",
      "95\t341\t0.0\t2\t337 4\n",
      "96\t747\t0.0\t2\t719 27 1\n",
      "97\t235\t0.0\t2\t225 8 2\n",
      "98\t171\t0.0\t2\t165 5 1\n",
      "99\t104\t0.0\t2\t103 1\n",
      "100\t106\t0.0\t2\t101 5\n",
      "101\t118\t0.0\t2\t116 2\n",
      "102\t94\t0.0\t2\t89 5\n",
      "103\t51\t0.0\t2\t50 0 1\n",
      "104\t23\t0.0\t2\t22 1\n",
      "105\t6\t0.0\t2\t6\n",
      "106\t6\t0.0\t2\t6\n",
      "107\t13\t0.0\t2\t13\n",
      "108\t13\t0.0\t2\t13\n",
      "109\t28\t0.0\t2\t28\n",
      "110\t31\t0.0\t2\t30 1\n",
      "111\t11\t0.0\t2\t11\n",
      "112\t25\t0.0\t2\t24 1\n",
      "113\t28\t0.0\t2\t26 2\n",
      "114\t24\t0.0\t2\t24\n",
      "115\t61\t0.0\t2\t60 1\n",
      "116\t70\t0.0\t2\t69 1\n",
      "117\t89\t0.0\t2\t86 1 2\n",
      "118\t134\t0.0\t2\t132 1 1\n",
      "119\t114\t0.0\t2\t109 4 1\n",
      "120\t73\t0.0\t2\t72 1\n",
      "121\t51\t0.0\t2\t50 0 1\n",
      "122\t90\t0.0\t2\t86 4\n",
      "123\t73\t0.0\t2\t66 7\n",
      "124\t70\t0.0\t2\t69 0 1\n",
      "125\t115\t0.0\t2\t112 2 1\n",
      "126\t103\t0.0\t2\t100 3\n",
      "127\t113\t0.0\t2\t104 6 3\n",
      "128\t190\t0.0\t2\t180 8 2\n",
      "129\t390\t0.0\t2\t381 7 2\n",
      "130\t503\t0.0\t2\t493 9 1\n",
      "131\t476\t0.0\t2\t464 11 1\n",
      "132\t556\t0.0\t2\t537 17 2\n",
      "133\t322\t0.0\t2\t318 4\n",
      "134\t582\t0.0\t2\t558 23 1\n",
      "135\t122\t0.0\t2\t119 3\n",
      "136\t142\t0.0\t2\t137 5\n",
      "137\t223\t0.0\t2\t214 8 1\n",
      "138\t105\t0.0\t2\t102 2 1\n",
      "139\t126\t0.0\t2\t122 4\n",
      "140\t279\t0.0\t2\t266 11 2\n",
      "141\t357\t0.0\t2\t68 124 165\n",
      "142\t6\t0.0\t2\t2 3 1\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "\n",
      "ks224\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -a GCGGTCGAGTTCTGGACCGA --minimum-length 11 -o fastq_5RACE/ks224-ctadpt_ctadpt3.fastq fastq_5RACE/ks224-ctadpt.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.40 s (11 us/read; 5.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  38,463\n",
      "Reads with adapters:                    31,302 (81.4%)\n",
      "Reads that were too short:               7,091 (18.4%)\n",
      "Reads written (passing filters):        31,372 (81.6%)\n",
      "\n",
      "Total basepairs processed:     5,424,453 bp\n",
      "Total written (filtered):      2,346,723 bp (43.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCGGTCGAGTTCTGGACCGA; Type: regular 3'; Length: 20; Trimmed: 31302 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 96.7%\n",
      "  C: 0.4%\n",
      "  G: 0.2%\n",
      "  T: 0.2%\n",
      "  none/other: 2.5%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"A\" extremely often.\n",
      "    The provided adapter sequence may be incomplete.\n",
      "    To fix the problem, add \"A\" to the beginning of the adapter sequence.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t214\t601.0\t0\t214\n",
      "4\t30\t150.2\t0\t30\n",
      "5\t58\t37.6\t0\t58\n",
      "6\t89\t9.4\t0\t89\n",
      "7\t88\t2.3\t0\t88\n",
      "8\t66\t0.6\t0\t66\n",
      "9\t60\t0.1\t0\t60\n",
      "10\t58\t0.0\t1\t57 1\n",
      "11\t56\t0.0\t1\t53 3\n",
      "12\t48\t0.0\t1\t47 1\n",
      "13\t35\t0.0\t1\t34 1\n",
      "14\t23\t0.0\t1\t23\n",
      "15\t15\t0.0\t1\t14 1\n",
      "16\t20\t0.0\t1\t20\n",
      "17\t46\t0.0\t1\t44 2\n",
      "18\t40\t0.0\t1\t39 1\n",
      "19\t32\t0.0\t1\t32\n",
      "20\t30\t0.0\t2\t29 1\n",
      "21\t35\t0.0\t2\t34 0 1\n",
      "22\t39\t0.0\t2\t35 4\n",
      "23\t31\t0.0\t2\t30 1\n",
      "24\t18\t0.0\t2\t17 1\n",
      "25\t18\t0.0\t2\t18\n",
      "26\t23\t0.0\t2\t23\n",
      "27\t53\t0.0\t2\t51 1 1\n",
      "28\t56\t0.0\t2\t55 1\n",
      "29\t151\t0.0\t2\t146 5\n",
      "30\t169\t0.0\t2\t163 6\n",
      "31\t182\t0.0\t2\t177 5\n",
      "32\t172\t0.0\t2\t167 5\n",
      "33\t91\t0.0\t2\t88 3\n",
      "34\t29\t0.0\t2\t27 2\n",
      "35\t23\t0.0\t2\t23\n",
      "36\t31\t0.0\t2\t29 2\n",
      "37\t30\t0.0\t2\t29 1\n",
      "38\t14\t0.0\t2\t12 1 1\n",
      "39\t21\t0.0\t2\t18 1 2\n",
      "40\t16\t0.0\t2\t16\n",
      "41\t12\t0.0\t2\t12\n",
      "42\t16\t0.0\t2\t15 1\n",
      "43\t20\t0.0\t2\t20\n",
      "44\t18\t0.0\t2\t18\n",
      "45\t29\t0.0\t2\t29\n",
      "46\t29\t0.0\t2\t28 1\n",
      "47\t47\t0.0\t2\t46 1\n",
      "48\t156\t0.0\t2\t154 2\n",
      "49\t262\t0.0\t2\t254 3 5\n",
      "50\t199\t0.0\t2\t196 3\n",
      "51\t110\t0.0\t2\t107 3\n",
      "52\t31\t0.0\t2\t29 1 1\n",
      "53\t37\t0.0\t2\t36 1\n",
      "54\t34\t0.0\t2\t34\n",
      "55\t58\t0.0\t2\t54 4\n",
      "56\t35\t0.0\t2\t34 0 1\n",
      "57\t53\t0.0\t2\t52 0 1\n",
      "58\t51\t0.0\t2\t48 1 2\n",
      "59\t64\t0.0\t2\t64\n",
      "60\t236\t0.0\t2\t230 6\n",
      "61\t411\t0.0\t2\t400 11\n",
      "62\t311\t0.0\t2\t303 8\n",
      "63\t84\t0.0\t2\t82 2\n",
      "64\t98\t0.0\t2\t93 5\n",
      "65\t172\t0.0\t2\t171 1\n",
      "66\t207\t0.0\t2\t201 6\n",
      "67\t185\t0.0\t2\t179 4 2\n",
      "68\t185\t0.0\t2\t182 2 1\n",
      "69\t244\t0.0\t2\t239 5\n",
      "70\t78\t0.0\t2\t78\n",
      "71\t74\t0.0\t2\t72 2\n",
      "72\t39\t0.0\t2\t39\n",
      "73\t67\t0.0\t2\t64 3\n",
      "74\t82\t0.0\t2\t80 2\n",
      "75\t82\t0.0\t2\t80 2\n",
      "76\t60\t0.0\t2\t57 3\n",
      "77\t67\t0.0\t2\t65 2\n",
      "78\t94\t0.0\t2\t90 2 2\n",
      "79\t266\t0.0\t2\t257 8 1\n",
      "80\t620\t0.0\t2\t601 16 3\n",
      "81\t864\t0.0\t2\t833 28 3\n",
      "82\t892\t0.0\t2\t863 24 5\n",
      "83\t815\t0.0\t2\t784 30 1\n",
      "84\t585\t0.0\t2\t570 14 1\n",
      "85\t567\t0.0\t2\t556 10 1\n",
      "86\t1164\t0.0\t2\t1139 23 2\n",
      "87\t484\t0.0\t2\t466 17 1\n",
      "88\t531\t0.0\t2\t509 18 4\n",
      "89\t363\t0.0\t2\t352 10 1\n",
      "90\t163\t0.0\t2\t161 2\n",
      "91\t128\t0.0\t2\t125 2 1\n",
      "92\t253\t0.0\t2\t249 2 2\n",
      "93\t453\t0.0\t2\t442 11\n",
      "94\t412\t0.0\t2\t405 6 1\n",
      "95\t743\t0.0\t2\t730 11 2\n",
      "96\t1212\t0.0\t2\t1182 27 3\n",
      "97\t660\t0.0\t2\t645 13 2\n",
      "98\t365\t0.0\t2\t356 9\n",
      "99\t222\t0.0\t2\t220 2\n",
      "100\t184\t0.0\t2\t179 5\n",
      "101\t350\t0.0\t2\t340 9 1\n",
      "102\t290\t0.0\t2\t289 1\n",
      "103\t167\t0.0\t2\t160 5 2\n",
      "104\t88\t0.0\t2\t85 2 1\n",
      "105\t10\t0.0\t2\t10\n",
      "106\t13\t0.0\t2\t12 0 1\n",
      "107\t13\t0.0\t2\t13\n",
      "108\t29\t0.0\t2\t28 1\n",
      "109\t32\t0.0\t2\t31 1\n",
      "110\t50\t0.0\t2\t49 1\n",
      "111\t40\t0.0\t2\t38 2\n",
      "112\t104\t0.0\t2\t102 2\n",
      "113\t45\t0.0\t2\t44 1\n",
      "114\t82\t0.0\t2\t80 2\n",
      "115\t111\t0.0\t2\t103 7 1\n",
      "116\t133\t0.0\t2\t128 5\n",
      "117\t155\t0.0\t2\t149 6\n",
      "118\t286\t0.0\t2\t276 9 1\n",
      "119\t313\t0.0\t2\t305 6 2\n",
      "120\t127\t0.0\t2\t123 4\n",
      "121\t106\t0.0\t2\t102 4\n",
      "122\t182\t0.0\t2\t176 5 1\n",
      "123\t130\t0.0\t2\t126 4\n",
      "124\t164\t0.0\t2\t154 10\n",
      "125\t237\t0.0\t2\t231 5 1\n",
      "126\t286\t0.0\t2\t278 7 1\n",
      "127\t221\t0.0\t2\t210 9 2\n",
      "128\t424\t0.0\t2\t415 8 1\n",
      "129\t950\t0.0\t2\t936 13 1\n",
      "130\t871\t0.0\t2\t853 17 1\n",
      "131\t1068\t0.0\t2\t1047 18 3\n",
      "132\t997\t0.0\t2\t979 17 1\n",
      "133\t679\t0.0\t2\t661 14 4\n",
      "134\t1193\t0.0\t2\t1171 19 3\n",
      "135\t264\t0.0\t2\t253 9 2\n",
      "136\t330\t0.0\t2\t319 11\n",
      "137\t566\t0.0\t2\t553 10 3\n",
      "138\t260\t0.0\t2\t251 6 3\n",
      "139\t322\t0.0\t2\t316 6\n",
      "140\t663\t0.0\t2\t635 25 3\n",
      "141\t752\t0.0\t2\t123 278 351\n",
      "142\t6\t0.0\t2\t2 4\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "\n",
      "ks225\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -a GCGGTCGAGTTCTGGACCGA --minimum-length 11 -o fastq_5RACE/ks225-ctadpt_ctadpt3.fastq fastq_5RACE/ks225-ctadpt.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.41 s (10 us/read; 6.07 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  41,469\n",
      "Reads with adapters:                    33,638 (81.1%)\n",
      "Reads that were too short:               7,056 (17.0%)\n",
      "Reads written (passing filters):        34,413 (83.0%)\n",
      "\n",
      "Total basepairs processed:     5,848,302 bp\n",
      "Total written (filtered):      2,591,316 bp (44.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCGGTCGAGTTCTGGACCGA; Type: regular 3'; Length: 20; Trimmed: 33638 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 97.0%\n",
      "  C: 0.4%\n",
      "  G: 0.2%\n",
      "  T: 0.2%\n",
      "  none/other: 2.3%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"A\" extremely often.\n",
      "    The provided adapter sequence may be incomplete.\n",
      "    To fix the problem, add \"A\" to the beginning of the adapter sequence.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t279\t648.0\t0\t279\n",
      "4\t27\t162.0\t0\t27\n",
      "5\t54\t40.5\t0\t54\n",
      "6\t114\t10.1\t0\t114\n",
      "7\t111\t2.5\t0\t111\n",
      "8\t82\t0.6\t0\t82\n",
      "9\t82\t0.2\t0\t82\n",
      "10\t68\t0.0\t1\t67 1\n",
      "11\t55\t0.0\t1\t54 1\n",
      "12\t51\t0.0\t1\t48 3\n",
      "13\t36\t0.0\t1\t33 3\n",
      "14\t29\t0.0\t1\t28 1\n",
      "15\t29\t0.0\t1\t28 1\n",
      "16\t30\t0.0\t1\t30\n",
      "17\t46\t0.0\t1\t45 1\n",
      "18\t62\t0.0\t1\t62\n",
      "19\t40\t0.0\t1\t38 2\n",
      "20\t30\t0.0\t2\t30\n",
      "21\t41\t0.0\t2\t40 1\n",
      "22\t26\t0.0\t2\t25 1\n",
      "23\t25\t0.0\t2\t25\n",
      "24\t27\t0.0\t2\t24 3\n",
      "25\t18\t0.0\t2\t18\n",
      "26\t29\t0.0\t2\t27 1 1\n",
      "27\t32\t0.0\t2\t31 0 1\n",
      "28\t78\t0.0\t2\t76 2\n",
      "29\t196\t0.0\t2\t189 6 1\n",
      "30\t205\t0.0\t2\t201 1 3\n",
      "31\t219\t0.0\t2\t215 3 1\n",
      "32\t225\t0.0\t2\t224 1\n",
      "33\t94\t0.0\t2\t92 2\n",
      "34\t39\t0.0\t2\t39\n",
      "35\t36\t0.0\t2\t35 0 1\n",
      "36\t41\t0.0\t2\t40 1\n",
      "37\t39\t0.0\t2\t37 1 1\n",
      "38\t13\t0.0\t2\t12 1\n",
      "39\t23\t0.0\t2\t22 1\n",
      "40\t23\t0.0\t2\t22 1\n",
      "41\t29\t0.0\t2\t28 0 1\n",
      "42\t19\t0.0\t2\t17 2\n",
      "43\t22\t0.0\t2\t22\n",
      "44\t22\t0.0\t2\t20 1 1\n",
      "45\t30\t0.0\t2\t30\n",
      "46\t33\t0.0\t2\t33\n",
      "47\t49\t0.0\t2\t47 1 1\n",
      "48\t151\t0.0\t2\t144 6 1\n",
      "49\t264\t0.0\t2\t261 3\n",
      "50\t235\t0.0\t2\t230 4 1\n",
      "51\t110\t0.0\t2\t105 5\n",
      "52\t34\t0.0\t2\t33 0 1\n",
      "53\t30\t0.0\t2\t29 1\n",
      "54\t48\t0.0\t2\t46 1 1\n",
      "55\t53\t0.0\t2\t51 1 1\n",
      "56\t40\t0.0\t2\t40\n",
      "57\t54\t0.0\t2\t54\n",
      "58\t53\t0.0\t2\t51 2\n",
      "59\t58\t0.0\t2\t58\n",
      "60\t296\t0.0\t2\t288 8\n",
      "61\t525\t0.0\t2\t512 13\n",
      "62\t322\t0.0\t2\t315 6 1\n",
      "63\t105\t0.0\t2\t103 1 1\n",
      "64\t105\t0.0\t2\t102 3\n",
      "65\t187\t0.0\t2\t183 3 1\n",
      "66\t219\t0.0\t2\t212 5 2\n",
      "67\t169\t0.0\t2\t168 1\n",
      "68\t177\t0.0\t2\t171 5 1\n",
      "69\t220\t0.0\t2\t214 6\n",
      "70\t88\t0.0\t2\t87 1\n",
      "71\t102\t0.0\t2\t101 0 1\n",
      "72\t54\t0.0\t2\t52 1 1\n",
      "73\t74\t0.0\t2\t70 4\n",
      "74\t90\t0.0\t2\t88 2\n",
      "75\t99\t0.0\t2\t93 5 1\n",
      "76\t62\t0.0\t2\t60 2\n",
      "77\t54\t0.0\t2\t51 2 1\n",
      "78\t108\t0.0\t2\t105 3\n",
      "79\t306\t0.0\t2\t301 4 1\n",
      "80\t656\t0.0\t2\t638 13 5\n",
      "81\t901\t0.0\t2\t874 26 1\n",
      "82\t843\t0.0\t2\t815 26 2\n",
      "83\t786\t0.0\t2\t758 25 3\n",
      "84\t665\t0.0\t2\t645 18 2\n",
      "85\t690\t0.0\t2\t672 17 1\n",
      "86\t1383\t0.0\t2\t1342 37 4\n",
      "87\t544\t0.0\t2\t534 10\n",
      "88\t626\t0.0\t2\t605 21\n",
      "89\t454\t0.0\t2\t443 10 1\n",
      "90\t205\t0.0\t2\t198 7\n",
      "91\t156\t0.0\t2\t155 1\n",
      "92\t303\t0.0\t2\t293 10\n",
      "93\t419\t0.0\t2\t409 9 1\n",
      "94\t473\t0.0\t2\t455 16 2\n",
      "95\t773\t0.0\t2\t752 18 3\n",
      "96\t1344\t0.0\t2\t1318 25 1\n",
      "97\t694\t0.0\t2\t681 11 2\n",
      "98\t449\t0.0\t2\t439 9 1\n",
      "99\t279\t0.0\t2\t269 8 2\n",
      "100\t235\t0.0\t2\t230 5\n",
      "101\t344\t0.0\t2\t335 8 1\n",
      "102\t276\t0.0\t2\t275 1\n",
      "103\t189\t0.0\t2\t187 2\n",
      "104\t72\t0.0\t2\t72\n",
      "105\t15\t0.0\t2\t15\n",
      "106\t13\t0.0\t2\t12 1\n",
      "107\t22\t0.0\t2\t20 0 2\n",
      "108\t20\t0.0\t2\t18 1 1\n",
      "109\t56\t0.0\t2\t56\n",
      "110\t59\t0.0\t2\t58 1\n",
      "111\t18\t0.0\t2\t18\n",
      "112\t53\t0.0\t2\t49 3 1\n",
      "113\t89\t0.0\t2\t87 0 2\n",
      "114\t80\t0.0\t2\t76 3 1\n",
      "115\t144\t0.0\t2\t138 4 2\n",
      "116\t116\t0.0\t2\t113 3\n",
      "117\t180\t0.0\t2\t174 5 1\n",
      "118\t331\t0.0\t2\t317 13 1\n",
      "119\t358\t0.0\t2\t347 7 4\n",
      "120\t165\t0.0\t2\t162 2 1\n",
      "121\t87\t0.0\t2\t83 2 2\n",
      "122\t169\t0.0\t2\t165 2 2\n",
      "123\t140\t0.0\t2\t134 6\n",
      "124\t139\t0.0\t2\t134 4 1\n",
      "125\t250\t0.0\t2\t238 9 3\n",
      "126\t264\t0.0\t2\t253 10 1\n",
      "127\t260\t0.0\t2\t254 6\n",
      "128\t462\t0.0\t2\t452 9 1\n",
      "129\t915\t0.0\t2\t887 22 6\n",
      "130\t1059\t0.0\t2\t1031 26 2\n",
      "131\t1008\t0.0\t2\t979 28 1\n",
      "132\t1058\t0.0\t2\t1027 28 3\n",
      "133\t638\t0.0\t2\t626 12\n",
      "134\t1246\t0.0\t2\t1206 38 2\n",
      "135\t305\t0.0\t2\t294 9 2\n",
      "136\t353\t0.0\t2\t340 13\n",
      "137\t534\t0.0\t2\t514 19 1\n",
      "138\t263\t0.0\t2\t258 4 1\n",
      "139\t298\t0.0\t2\t288 8 2\n",
      "140\t610\t0.0\t2\t586 20 4\n",
      "141\t748\t0.0\t2\t96 287 365\n",
      "142\t4\t0.0\t2\t0 2 2\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "\n",
      "ks230\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -a GCGGTCGAGTTCTGGACCGA --minimum-length 11 -o fastq_5RACE/ks230-ctadpt_ctadpt3.fastq fastq_5RACE/ks230-ctadpt.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.32 s (9 us/read; 6.39 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  34,544\n",
      "Reads with adapters:                    28,166 (81.5%)\n",
      "Reads that were too short:               2,982 (8.6%)\n",
      "Reads written (passing filters):        31,562 (91.4%)\n",
      "\n",
      "Total basepairs processed:     4,873,104 bp\n",
      "Total written (filtered):      2,544,027 bp (52.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCGGTCGAGTTCTGGACCGA; Type: regular 3'; Length: 20; Trimmed: 28166 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 96.3%\n",
      "  C: 0.6%\n",
      "  G: 0.2%\n",
      "  T: 0.2%\n",
      "  none/other: 2.7%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"A\" extremely often.\n",
      "    The provided adapter sequence may be incomplete.\n",
      "    To fix the problem, add \"A\" to the beginning of the adapter sequence.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t257\t539.8\t0\t257\n",
      "4\t18\t134.9\t0\t18\n",
      "5\t31\t33.7\t0\t31\n",
      "6\t45\t8.4\t0\t45\n",
      "7\t63\t2.1\t0\t63\n",
      "8\t41\t0.5\t0\t41\n",
      "9\t140\t0.1\t0\t140\n",
      "10\t67\t0.0\t1\t65 2\n",
      "11\t70\t0.0\t1\t69 1\n",
      "12\t90\t0.0\t1\t88 2\n",
      "13\t104\t0.0\t1\t101 3\n",
      "14\t51\t0.0\t1\t51\n",
      "15\t59\t0.0\t1\t58 1\n",
      "16\t30\t0.0\t1\t30\n",
      "17\t37\t0.0\t1\t36 1\n",
      "18\t57\t0.0\t1\t56 1\n",
      "19\t60\t0.0\t1\t60\n",
      "20\t48\t0.0\t2\t44 4\n",
      "21\t35\t0.0\t2\t32 3\n",
      "22\t34\t0.0\t2\t34\n",
      "23\t52\t0.0\t2\t52\n",
      "24\t60\t0.0\t2\t60\n",
      "25\t69\t0.0\t2\t67 1 1\n",
      "26\t57\t0.0\t2\t57\n",
      "27\t51\t0.0\t2\t49 1 1\n",
      "28\t37\t0.0\t2\t37\n",
      "29\t29\t0.0\t2\t26 3\n",
      "30\t18\t0.0\t2\t17 0 1\n",
      "31\t28\t0.0\t2\t24 2 2\n",
      "32\t19\t0.0\t2\t19\n",
      "33\t59\t0.0\t2\t59\n",
      "34\t65\t0.0\t2\t65\n",
      "35\t54\t0.0\t2\t54\n",
      "36\t107\t0.0\t2\t105 2\n",
      "37\t78\t0.0\t2\t77 0 1\n",
      "38\t38\t0.0\t2\t38\n",
      "39\t31\t0.0\t2\t27 3 1\n",
      "40\t97\t0.0\t2\t94 3\n",
      "41\t65\t0.0\t2\t62 2 1\n",
      "42\t59\t0.0\t2\t58 0 1\n",
      "43\t44\t0.0\t2\t42 0 2\n",
      "44\t31\t0.0\t2\t30 0 1\n",
      "45\t22\t0.0\t2\t22\n",
      "46\t52\t0.0\t2\t50 1 1\n",
      "47\t94\t0.0\t2\t92 1 1\n",
      "48\t124\t0.0\t2\t120 3 1\n",
      "49\t134\t0.0\t2\t129 5\n",
      "50\t92\t0.0\t2\t89 3\n",
      "51\t90\t0.0\t2\t89 1\n",
      "52\t70\t0.0\t2\t68 1 1\n",
      "53\t68\t0.0\t2\t66 1 1\n",
      "54\t54\t0.0\t2\t51 3\n",
      "55\t51\t0.0\t2\t51\n",
      "56\t52\t0.0\t2\t51 1\n",
      "57\t62\t0.0\t2\t61 0 1\n",
      "58\t84\t0.0\t2\t81 2 1\n",
      "59\t109\t0.0\t2\t107 2\n",
      "60\t112\t0.0\t2\t111 1\n",
      "61\t266\t0.0\t2\t257 9\n",
      "62\t104\t0.0\t2\t99 4 1\n",
      "63\t2017\t0.0\t2\t1959 55 3\n",
      "64\t5946\t0.0\t2\t5796 126 24\n",
      "65\t155\t0.0\t2\t147 6 2\n",
      "66\t104\t0.0\t2\t103 1\n",
      "67\t177\t0.0\t2\t175 2\n",
      "68\t242\t0.0\t2\t235 7\n",
      "69\t82\t0.0\t2\t76 6\n",
      "70\t114\t0.0\t2\t113 1\n",
      "71\t265\t0.0\t2\t254 10 1\n",
      "72\t101\t0.0\t2\t98 3\n",
      "73\t151\t0.0\t2\t147 4\n",
      "74\t282\t0.0\t2\t274 6 2\n",
      "75\t310\t0.0\t2\t296 13 1\n",
      "76\t359\t0.0\t2\t348 8 3\n",
      "77\t241\t0.0\t2\t234 6 1\n",
      "78\t212\t0.0\t2\t203 8 1\n",
      "79\t351\t0.0\t2\t338 11 2\n",
      "80\t355\t0.0\t2\t350 5\n",
      "81\t238\t0.0\t2\t232 5 1\n",
      "82\t230\t0.0\t2\t224 5 1\n",
      "83\t293\t0.0\t2\t287 5 1\n",
      "84\t267\t0.0\t2\t261 5 1\n",
      "85\t397\t0.0\t2\t383 13 1\n",
      "86\t533\t0.0\t2\t518 14 1\n",
      "87\t250\t0.0\t2\t242 8\n",
      "88\t238\t0.0\t2\t230 5 3\n",
      "89\t179\t0.0\t2\t174 5\n",
      "90\t126\t0.0\t2\t122 3 1\n",
      "91\t82\t0.0\t2\t80 1 1\n",
      "92\t139\t0.0\t2\t135 4\n",
      "93\t210\t0.0\t2\t206 4\n",
      "94\t244\t0.0\t2\t238 5 1\n",
      "95\t436\t0.0\t2\t427 8 1\n",
      "96\t787\t0.0\t2\t769 17 1\n",
      "97\t367\t0.0\t2\t356 11\n",
      "98\t233\t0.0\t2\t228 5\n",
      "99\t137\t0.0\t2\t131 4 2\n",
      "100\t119\t0.0\t2\t116 3\n",
      "101\t134\t0.0\t2\t132 2\n",
      "102\t82\t0.0\t2\t81 1\n",
      "103\t96\t0.0\t2\t94 2\n",
      "104\t38\t0.0\t2\t38\n",
      "105\t11\t0.0\t2\t11\n",
      "106\t9\t0.0\t2\t9\n",
      "107\t10\t0.0\t2\t10\n",
      "108\t27\t0.0\t2\t26 1\n",
      "109\t30\t0.0\t2\t30\n",
      "110\t47\t0.0\t2\t44 3\n",
      "111\t31\t0.0\t2\t30 1\n",
      "112\t19\t0.0\t2\t19\n",
      "113\t18\t0.0\t2\t17 1\n",
      "114\t38\t0.0\t2\t36 2\n",
      "115\t69\t0.0\t2\t65 4\n",
      "116\t126\t0.0\t2\t125 1\n",
      "117\t139\t0.0\t2\t135 4\n",
      "118\t221\t0.0\t2\t213 7 1\n",
      "119\t210\t0.0\t2\t204 6\n",
      "120\t143\t0.0\t2\t138 5\n",
      "121\t92\t0.0\t2\t90 2\n",
      "122\t122\t0.0\t2\t117 5\n",
      "123\t80\t0.0\t2\t79 1\n",
      "124\t48\t0.0\t2\t48\n",
      "125\t152\t0.0\t2\t149 3\n",
      "126\t86\t0.0\t2\t85 0 1\n",
      "127\t137\t0.0\t2\t132 5\n",
      "128\t129\t0.0\t2\t124 5\n",
      "129\t341\t0.0\t2\t334 5 2\n",
      "130\t1393\t0.0\t2\t1347 42 4\n",
      "131\t716\t0.0\t2\t704 12\n",
      "132\t475\t0.0\t2\t463 12\n",
      "133\t216\t0.0\t2\t211 3 2\n",
      "134\t359\t0.0\t2\t349 8 2\n",
      "135\t76\t0.0\t2\t75 0 1\n",
      "136\t39\t0.0\t2\t37 2\n",
      "137\t49\t0.0\t2\t47 2\n",
      "138\t43\t0.0\t2\t41 2\n",
      "139\t53\t0.0\t2\t53\n",
      "140\t246\t0.0\t2\t211 33 2\n",
      "141\t721\t0.0\t2\t506 159 56\n",
      "142\t1\t0.0\t2\t1\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "\n",
      "ks231\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -a GCGGTCGAGTTCTGGACCGA --minimum-length 11 -o fastq_5RACE/ks231-ctadpt_ctadpt3.fastq fastq_5RACE/ks231-ctadpt.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.34 s (9 us/read; 6.49 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  37,073\n",
      "Reads with adapters:                    29,805 (80.4%)\n",
      "Reads that were too short:               3,866 (10.4%)\n",
      "Reads written (passing filters):        33,207 (89.6%)\n",
      "\n",
      "Total basepairs processed:     5,229,639 bp\n",
      "Total written (filtered):      2,611,429 bp (49.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCGGTCGAGTTCTGGACCGA; Type: regular 3'; Length: 20; Trimmed: 29805 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 95.1%\n",
      "  C: 0.9%\n",
      "  G: 0.3%\n",
      "  T: 0.1%\n",
      "  none/other: 3.6%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"A\" extremely often.\n",
      "    The provided adapter sequence may be incomplete.\n",
      "    To fix the problem, add \"A\" to the beginning of the adapter sequence.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t333\t579.3\t0\t333\n",
      "4\t26\t144.8\t0\t26\n",
      "5\t44\t36.2\t0\t44\n",
      "6\t85\t9.1\t0\t85\n",
      "7\t73\t2.3\t0\t73\n",
      "8\t40\t0.6\t0\t40\n",
      "9\t125\t0.1\t0\t125\n",
      "10\t88\t0.0\t1\t87 1\n",
      "11\t116\t0.0\t1\t115 1\n",
      "12\t99\t0.0\t1\t99\n",
      "13\t163\t0.0\t1\t159 4\n",
      "14\t74\t0.0\t1\t73 1\n",
      "15\t93\t0.0\t1\t91 2\n",
      "16\t54\t0.0\t1\t54\n",
      "17\t84\t0.0\t1\t82 2\n",
      "18\t76\t0.0\t1\t72 4\n",
      "19\t84\t0.0\t1\t81 2 1\n",
      "20\t60\t0.0\t2\t58 1 1\n",
      "21\t59\t0.0\t2\t58 0 1\n",
      "22\t53\t0.0\t2\t51 2\n",
      "23\t82\t0.0\t2\t79 2 1\n",
      "24\t78\t0.0\t2\t78\n",
      "25\t90\t0.0\t2\t87 3\n",
      "26\t118\t0.0\t2\t117 1\n",
      "27\t84\t0.0\t2\t82 2\n",
      "28\t60\t0.0\t2\t57 3\n",
      "29\t45\t0.0\t2\t44 1\n",
      "30\t22\t0.0\t2\t21 1\n",
      "31\t27\t0.0\t2\t25 2\n",
      "32\t44\t0.0\t2\t43 1\n",
      "33\t78\t0.0\t2\t73 4 1\n",
      "34\t93\t0.0\t2\t91 1 1\n",
      "35\t79\t0.0\t2\t79\n",
      "36\t183\t0.0\t2\t180 3\n",
      "37\t88\t0.0\t2\t86 1 1\n",
      "38\t46\t0.0\t2\t45 0 1\n",
      "39\t48\t0.0\t2\t45 0 3\n",
      "40\t88\t0.0\t2\t84 4\n",
      "41\t86\t0.0\t2\t83 2 1\n",
      "42\t90\t0.0\t2\t88 2\n",
      "43\t53\t0.0\t2\t49 2 2\n",
      "44\t34\t0.0\t2\t30 4\n",
      "45\t51\t0.0\t2\t50 1\n",
      "46\t99\t0.0\t2\t94 3 2\n",
      "47\t112\t0.0\t2\t109 3\n",
      "48\t209\t0.0\t2\t205 3 1\n",
      "49\t219\t0.0\t2\t215 3 1\n",
      "50\t162\t0.0\t2\t161 1\n",
      "51\t125\t0.0\t2\t124 1\n",
      "52\t88\t0.0\t2\t81 5 2\n",
      "53\t98\t0.0\t2\t95 3\n",
      "54\t56\t0.0\t2\t55 1\n",
      "55\t67\t0.0\t2\t65 2\n",
      "56\t49\t0.0\t2\t49\n",
      "57\t58\t0.0\t2\t53 3 2\n",
      "58\t72\t0.0\t2\t69 3\n",
      "59\t123\t0.0\t2\t122 1\n",
      "60\t131\t0.0\t2\t128 3\n",
      "61\t156\t0.0\t2\t148 8\n",
      "62\t86\t0.0\t2\t83 2 1\n",
      "63\t978\t0.0\t2\t956 20 2\n",
      "64\t1685\t0.0\t2\t1645 36 4\n",
      "65\t91\t0.0\t2\t88 3\n",
      "66\t99\t0.0\t2\t95 4\n",
      "67\t105\t0.0\t2\t100 4 1\n",
      "68\t107\t0.0\t2\t104 3\n",
      "69\t83\t0.0\t2\t79 3 1\n",
      "70\t184\t0.0\t2\t180 3 1\n",
      "71\t203\t0.0\t2\t197 6\n",
      "72\t129\t0.0\t2\t126 3\n",
      "73\t254\t0.0\t2\t250 4\n",
      "74\t455\t0.0\t2\t445 10\n",
      "75\t508\t0.0\t2\t498 8 2\n",
      "76\t542\t0.0\t2\t527 14 1\n",
      "77\t318\t0.0\t2\t309 8 1\n",
      "78\t319\t0.0\t2\t307 12\n",
      "79\t507\t0.0\t2\t489 15 3\n",
      "80\t558\t0.0\t2\t544 13 1\n",
      "81\t355\t0.0\t2\t346 8 1\n",
      "82\t368\t0.0\t2\t356 10 2\n",
      "83\t372\t0.0\t2\t355 14 3\n",
      "84\t363\t0.0\t2\t349 10 4\n",
      "85\t439\t0.0\t2\t430 8 1\n",
      "86\t592\t0.0\t2\t566 26\n",
      "87\t379\t0.0\t2\t371 8\n",
      "88\t399\t0.0\t2\t388 11\n",
      "89\t328\t0.0\t2\t313 15\n",
      "90\t183\t0.0\t2\t178 4 1\n",
      "91\t134\t0.0\t2\t133 1\n",
      "92\t215\t0.0\t2\t209 3 3\n",
      "93\t297\t0.0\t2\t287 8 2\n",
      "94\t325\t0.0\t2\t316 8 1\n",
      "95\t614\t0.0\t2\t599 14 1\n",
      "96\t1045\t0.0\t2\t1019 24 2\n",
      "97\t624\t0.0\t2\t603 18 3\n",
      "98\t363\t0.0\t2\t359 4\n",
      "99\t217\t0.0\t2\t212 5\n",
      "100\t213\t0.0\t2\t206 6 1\n",
      "101\t190\t0.0\t2\t185 4 1\n",
      "102\t136\t0.0\t2\t133 3\n",
      "103\t151\t0.0\t2\t150 1\n",
      "104\t57\t0.0\t2\t54 3\n",
      "105\t15\t0.0\t2\t14 1\n",
      "106\t15\t0.0\t2\t14 1\n",
      "107\t16\t0.0\t2\t16\n",
      "108\t28\t0.0\t2\t28\n",
      "109\t44\t0.0\t2\t43 1\n",
      "110\t57\t0.0\t2\t53 3 1\n",
      "111\t40\t0.0\t2\t40\n",
      "112\t57\t0.0\t2\t56 1\n",
      "113\t38\t0.0\t2\t37 1\n",
      "114\t73\t0.0\t2\t69 4\n",
      "115\t101\t0.0\t2\t98 3\n",
      "116\t178\t0.0\t2\t171 5 2\n",
      "117\t258\t0.0\t2\t252 5 1\n",
      "118\t311\t0.0\t2\t305 6\n",
      "119\t304\t0.0\t2\t292 11 1\n",
      "120\t200\t0.0\t2\t191 9\n",
      "121\t119\t0.0\t2\t115 4\n",
      "122\t152\t0.0\t2\t144 7 1\n",
      "123\t124\t0.0\t2\t121 3\n",
      "124\t93\t0.0\t2\t89 3 1\n",
      "125\t193\t0.0\t2\t190 2 1\n",
      "126\t100\t0.0\t2\t97 3\n",
      "127\t171\t0.0\t2\t168 3\n",
      "128\t172\t0.0\t2\t167 3 2\n",
      "129\t425\t0.0\t2\t410 15\n",
      "130\t1743\t0.0\t2\t1693 47 3\n",
      "131\t859\t0.0\t2\t832 24 3\n",
      "132\t685\t0.0\t2\t668 17\n",
      "133\t289\t0.0\t2\t279 6 4\n",
      "134\t386\t0.0\t2\t379 6 1\n",
      "135\t77\t0.0\t2\t71 3 3\n",
      "136\t60\t0.0\t2\t59 1\n",
      "137\t61\t0.0\t2\t57 3 1\n",
      "138\t58\t0.0\t2\t53 3 2\n",
      "139\t111\t0.0\t2\t106 4 1\n",
      "140\t298\t0.0\t2\t239 56 3\n",
      "141\t994\t0.0\t2\t600 300 94\n",
      "142\t10\t0.0\t2\t3 7\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "\n",
      "ks232\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -a GCGGTCGAGTTCTGGACCGA --minimum-length 11 -o fastq_5RACE/ks232-ctadpt_ctadpt3.fastq fastq_5RACE/ks232-ctadpt.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.35 s (10 us/read; 6.28 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  36,091\n",
      "Reads with adapters:                    27,830 (77.1%)\n",
      "Reads that were too short:               3,334 (9.2%)\n",
      "Reads written (passing filters):        32,757 (90.8%)\n",
      "\n",
      "Total basepairs processed:     5,091,414 bp\n",
      "Total written (filtered):      2,612,236 bp (51.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCGGTCGAGTTCTGGACCGA; Type: regular 3'; Length: 20; Trimmed: 27830 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 97.6%\n",
      "  C: 1.1%\n",
      "  G: 0.1%\n",
      "  T: 0.2%\n",
      "  none/other: 1.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"A\" extremely often.\n",
      "    The provided adapter sequence may be incomplete.\n",
      "    To fix the problem, add \"A\" to the beginning of the adapter sequence.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t396\t563.9\t0\t396\n",
      "4\t36\t141.0\t0\t36\n",
      "5\t49\t35.2\t0\t49\n",
      "6\t104\t8.8\t0\t104\n",
      "7\t62\t2.2\t0\t62\n",
      "8\t50\t0.6\t0\t50\n",
      "9\t37\t0.1\t0\t37\n",
      "10\t72\t0.0\t1\t71 1\n",
      "11\t100\t0.0\t1\t96 4\n",
      "12\t113\t0.0\t1\t110 3\n",
      "13\t170\t0.0\t1\t168 2\n",
      "14\t109\t0.0\t1\t107 2\n",
      "15\t108\t0.0\t1\t105 3\n",
      "16\t75\t0.0\t1\t75\n",
      "17\t85\t0.0\t1\t83 2\n",
      "18\t110\t0.0\t1\t110\n",
      "19\t85\t0.0\t1\t82 2 1\n",
      "20\t79\t0.0\t2\t75 4\n",
      "21\t56\t0.0\t2\t55 1\n",
      "22\t57\t0.0\t2\t56 1\n",
      "23\t82\t0.0\t2\t79 2 1\n",
      "24\t94\t0.0\t2\t92 2\n",
      "25\t110\t0.0\t2\t109 1\n",
      "26\t123\t0.0\t2\t122 1\n",
      "27\t116\t0.0\t2\t109 4 3\n",
      "28\t60\t0.0\t2\t56 0 4\n",
      "29\t52\t0.0\t2\t52\n",
      "30\t36\t0.0\t2\t34 2\n",
      "31\t30\t0.0\t2\t29 1\n",
      "32\t29\t0.0\t2\t29\n",
      "33\t71\t0.0\t2\t71\n",
      "34\t84\t0.0\t2\t84\n",
      "35\t103\t0.0\t2\t103\n",
      "36\t69\t0.0\t2\t65 4\n",
      "37\t47\t0.0\t2\t46 1\n",
      "38\t49\t0.0\t2\t49\n",
      "39\t56\t0.0\t2\t55 1\n",
      "40\t57\t0.0\t2\t55 1 1\n",
      "41\t92\t0.0\t2\t90 1 1\n",
      "42\t119\t0.0\t2\t119\n",
      "43\t74\t0.0\t2\t71 3\n",
      "44\t26\t0.0\t2\t26\n",
      "45\t36\t0.0\t2\t35 1\n",
      "46\t96\t0.0\t2\t93 2 1\n",
      "47\t139\t0.0\t2\t137 1 1\n",
      "48\t209\t0.0\t2\t203 6\n",
      "49\t279\t0.0\t2\t276 3\n",
      "50\t179\t0.0\t2\t177 2\n",
      "51\t147\t0.0\t2\t145 1 1\n",
      "52\t117\t0.0\t2\t114 3\n",
      "53\t96\t0.0\t2\t92 4\n",
      "54\t61\t0.0\t2\t57 4\n",
      "55\t53\t0.0\t2\t51 2\n",
      "56\t44\t0.0\t2\t43 1\n",
      "57\t54\t0.0\t2\t50 3 1\n",
      "58\t70\t0.0\t2\t66 4\n",
      "59\t114\t0.0\t2\t109 3 2\n",
      "60\t156\t0.0\t2\t152 3 1\n",
      "61\t82\t0.0\t2\t82\n",
      "62\t57\t0.0\t2\t57\n",
      "63\t51\t0.0\t2\t49 2\n",
      "64\t68\t0.0\t2\t65 3\n",
      "65\t46\t0.0\t2\t46\n",
      "66\t48\t0.0\t2\t46 1 1\n",
      "67\t48\t0.0\t2\t43 4 1\n",
      "68\t49\t0.0\t2\t48 1\n",
      "69\t96\t0.0\t2\t90 6\n",
      "70\t178\t0.0\t2\t171 6 1\n",
      "71\t151\t0.0\t2\t148 3\n",
      "72\t143\t0.0\t2\t140 3\n",
      "73\t310\t0.0\t2\t301 7 2\n",
      "74\t481\t0.0\t2\t471 10\n",
      "75\t560\t0.0\t2\t544 13 3\n",
      "76\t588\t0.0\t2\t574 13 1\n",
      "77\t412\t0.0\t2\t397 15\n",
      "78\t411\t0.0\t2\t402 9\n",
      "79\t592\t0.0\t2\t572 17 3\n",
      "80\t642\t0.0\t2\t628 13 1\n",
      "81\t444\t0.0\t2\t432 11 1\n",
      "82\t417\t0.0\t2\t407 8 2\n",
      "83\t441\t0.0\t2\t427 13 1\n",
      "84\t436\t0.0\t2\t424 11 1\n",
      "85\t361\t0.0\t2\t349 10 2\n",
      "86\t512\t0.0\t2\t498 13 1\n",
      "87\t456\t0.0\t2\t443 12 1\n",
      "88\t483\t0.0\t2\t464 18 1\n",
      "89\t365\t0.0\t2\t356 8 1\n",
      "90\t230\t0.0\t2\t226 4\n",
      "91\t139\t0.0\t2\t137 2\n",
      "92\t232\t0.0\t2\t228 3 1\n",
      "93\t377\t0.0\t2\t368 8 1\n",
      "94\t346\t0.0\t2\t337 9\n",
      "95\t545\t0.0\t2\t534 10 1\n",
      "96\t840\t0.0\t2\t820 19 1\n",
      "97\t665\t0.0\t2\t650 11 4\n",
      "98\t442\t0.0\t2\t431 9 2\n",
      "99\t239\t0.0\t2\t232 6 1\n",
      "100\t235\t0.0\t2\t232 3\n",
      "101\t237\t0.0\t2\t230 5 2\n",
      "102\t144\t0.0\t2\t139 4 1\n",
      "103\t137\t0.0\t2\t135 2\n",
      "104\t64\t0.0\t2\t63 1\n",
      "105\t23\t0.0\t2\t22 1\n",
      "106\t14\t0.0\t2\t14\n",
      "107\t24\t0.0\t2\t22 2\n",
      "108\t29\t0.0\t2\t28 1\n",
      "109\t33\t0.0\t2\t33\n",
      "110\t39\t0.0\t2\t39\n",
      "111\t62\t0.0\t2\t62\n",
      "112\t48\t0.0\t2\t48\n",
      "113\t43\t0.0\t2\t43\n",
      "114\t63\t0.0\t2\t59 2 2\n",
      "115\t87\t0.0\t2\t85 1 1\n",
      "116\t184\t0.0\t2\t179 5\n",
      "117\t148\t0.0\t2\t143 4 1\n",
      "118\t325\t0.0\t2\t316 6 3\n",
      "119\t333\t0.0\t2\t320 9 4\n",
      "120\t195\t0.0\t2\t190 3 2\n",
      "121\t134\t0.0\t2\t127 6 1\n",
      "122\t148\t0.0\t2\t139 4 5\n",
      "123\t123\t0.0\t2\t119 4\n",
      "124\t92\t0.0\t2\t89 3\n",
      "125\t148\t0.0\t2\t143 4 1\n",
      "126\t115\t0.0\t2\t111 4\n",
      "127\t204\t0.0\t2\t197 5 2\n",
      "128\t150\t0.0\t2\t147 2 1\n",
      "129\t447\t0.0\t2\t438 7 2\n",
      "130\t2091\t0.0\t2\t2032 52 7\n",
      "131\t1155\t0.0\t2\t1133 21 1\n",
      "132\t728\t0.0\t2\t711 17\n",
      "133\t254\t0.0\t2\t251 3\n",
      "134\t424\t0.0\t2\t415 8 1\n",
      "135\t61\t0.0\t2\t56 4 1\n",
      "136\t46\t0.0\t2\t44 1 1\n",
      "137\t69\t0.0\t2\t67 0 2\n",
      "138\t44\t0.0\t2\t42 1 1\n",
      "139\t87\t0.0\t2\t86 0 1\n",
      "140\t209\t0.0\t2\t199 5 5\n",
      "141\t265\t0.0\t2\t43 161 61\n",
      "142\t4\t0.0\t2\t0 4\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "\n",
      "ks233\n",
      "This is cutadapt 1.16 with Python 2.7.15\n",
      "Command line parameters: -a GCGGTCGAGTTCTGGACCGA --minimum-length 11 -o fastq_5RACE/ks233-ctadpt_ctadpt3.fastq fastq_5RACE/ks233-ctadpt.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.39 s (10 us/read; 6.18 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  39,853\n",
      "Reads with adapters:                    30,098 (75.5%)\n",
      "Reads that were too short:               3,369 (8.5%)\n",
      "Reads written (passing filters):        36,484 (91.5%)\n",
      "\n",
      "Total basepairs processed:     5,622,070 bp\n",
      "Total written (filtered):      2,959,291 bp (52.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GCGGTCGAGTTCTGGACCGA; Type: regular 3'; Length: 20; Trimmed: 30098 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 97.7%\n",
      "  C: 1.1%\n",
      "  G: 0.2%\n",
      "  T: 0.2%\n",
      "  none/other: 0.8%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"A\" extremely often.\n",
      "    The provided adapter sequence may be incomplete.\n",
      "    To fix the problem, add \"A\" to the beginning of the adapter sequence.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t475\t622.7\t0\t475\n",
      "4\t26\t155.7\t0\t26\n",
      "5\t61\t38.9\t0\t61\n",
      "6\t97\t9.7\t0\t97\n",
      "7\t77\t2.4\t0\t77\n",
      "8\t58\t0.6\t0\t58\n",
      "9\t42\t0.2\t0\t42\n",
      "10\t83\t0.0\t1\t83\n",
      "11\t119\t0.0\t1\t118 1\n",
      "12\t127\t0.0\t1\t125 2\n",
      "13\t193\t0.0\t1\t189 4\n",
      "14\t121\t0.0\t1\t120 1\n",
      "15\t99\t0.0\t1\t99\n",
      "16\t66\t0.0\t1\t64 2\n",
      "17\t92\t0.0\t1\t89 3\n",
      "18\t115\t0.0\t1\t115\n",
      "19\t75\t0.0\t1\t75\n",
      "20\t82\t0.0\t2\t80 2\n",
      "21\t66\t0.0\t2\t63 1 2\n",
      "22\t60\t0.0\t2\t56 3 1\n",
      "23\t107\t0.0\t2\t103 3 1\n",
      "24\t98\t0.0\t2\t95 2 1\n",
      "25\t138\t0.0\t2\t133 4 1\n",
      "26\t136\t0.0\t2\t129 7\n",
      "27\t117\t0.0\t2\t114 3\n",
      "28\t73\t0.0\t2\t70 2 1\n",
      "29\t59\t0.0\t2\t53 5 1\n",
      "30\t33\t0.0\t2\t33\n",
      "31\t40\t0.0\t2\t37 3\n",
      "32\t46\t0.0\t2\t45 1\n",
      "33\t95\t0.0\t2\t92 2 1\n",
      "34\t96\t0.0\t2\t90 4 2\n",
      "35\t105\t0.0\t2\t103 2\n",
      "36\t80\t0.0\t2\t78 2\n",
      "37\t39\t0.0\t2\t38 1\n",
      "38\t42\t0.0\t2\t41 0 1\n",
      "39\t45\t0.0\t2\t42 2 1\n",
      "40\t64\t0.0\t2\t63 1\n",
      "41\t103\t0.0\t2\t101 2\n",
      "42\t112\t0.0\t2\t108 4\n",
      "43\t58\t0.0\t2\t54 1 3\n",
      "44\t26\t0.0\t2\t26\n",
      "45\t47\t0.0\t2\t45 1 1\n",
      "46\t104\t0.0\t2\t101 3\n",
      "47\t154\t0.0\t2\t150 3 1\n",
      "48\t252\t0.0\t2\t249 3\n",
      "49\t295\t0.0\t2\t285 10\n",
      "50\t185\t0.0\t2\t181 2 2\n",
      "51\t170\t0.0\t2\t165 5\n",
      "52\t144\t0.0\t2\t137 6 1\n",
      "53\t118\t0.0\t2\t117 1\n",
      "54\t75\t0.0\t2\t73 1 1\n",
      "55\t57\t0.0\t2\t56 1\n",
      "56\t48\t0.0\t2\t48\n",
      "57\t64\t0.0\t2\t62 2\n",
      "58\t67\t0.0\t2\t61 4 2\n",
      "59\t135\t0.0\t2\t134 1\n",
      "60\t166\t0.0\t2\t164 2\n",
      "61\t85\t0.0\t2\t83 2\n",
      "62\t68\t0.0\t2\t65 3\n",
      "63\t58\t0.0\t2\t57 1\n",
      "64\t68\t0.0\t2\t67 1\n",
      "65\t72\t0.0\t2\t68 2 2\n",
      "66\t66\t0.0\t2\t65 1\n",
      "67\t51\t0.0\t2\t49 2\n",
      "68\t60\t0.0\t2\t55 4 1\n",
      "69\t82\t0.0\t2\t79 3\n",
      "70\t179\t0.0\t2\t176 3\n",
      "71\t172\t0.0\t2\t165 6 1\n",
      "72\t185\t0.0\t2\t178 7\n",
      "73\t344\t0.0\t2\t330 9 5\n",
      "74\t592\t0.0\t2\t582 8 2\n",
      "75\t611\t0.0\t2\t592 14 5\n",
      "76\t664\t0.0\t2\t645 16 3\n",
      "77\t389\t0.0\t2\t378 7 4\n",
      "78\t450\t0.0\t2\t435 12 3\n",
      "79\t637\t0.0\t2\t615 20 2\n",
      "80\t691\t0.0\t2\t676 12 3\n",
      "81\t448\t0.0\t2\t436 11 1\n",
      "82\t411\t0.0\t2\t389 21 1\n",
      "83\t484\t0.0\t2\t466 16 2\n",
      "84\t471\t0.0\t2\t447 23 1\n",
      "85\t431\t0.0\t2\t412 18 1\n",
      "86\t587\t0.0\t2\t569 16 2\n",
      "87\t458\t0.0\t2\t447 8 3\n",
      "88\t457\t0.0\t2\t437 18 2\n",
      "89\t378\t0.0\t2\t369 6 3\n",
      "90\t243\t0.0\t2\t235 8\n",
      "91\t140\t0.0\t2\t140\n",
      "92\t246\t0.0\t2\t240 5 1\n",
      "93\t379\t0.0\t2\t374 4 1\n",
      "94\t395\t0.0\t2\t389 5 1\n",
      "95\t583\t0.0\t2\t573 9 1\n",
      "96\t957\t0.0\t2\t928 23 6\n",
      "97\t771\t0.0\t2\t759 8 4\n",
      "98\t508\t0.0\t2\t497 10 1\n",
      "99\t265\t0.0\t2\t259 6\n",
      "100\t236\t0.0\t2\t229 7\n",
      "101\t259\t0.0\t2\t254 3 2\n",
      "102\t162\t0.0\t2\t159 3\n",
      "103\t159\t0.0\t2\t155 2 2\n",
      "104\t79\t0.0\t2\t75 4\n",
      "105\t15\t0.0\t2\t14 0 1\n",
      "106\t8\t0.0\t2\t7 0 1\n",
      "107\t22\t0.0\t2\t22\n",
      "108\t23\t0.0\t2\t23\n",
      "109\t38\t0.0\t2\t38\n",
      "110\t60\t0.0\t2\t59 1\n",
      "111\t57\t0.0\t2\t54 3\n",
      "112\t57\t0.0\t2\t53 4\n",
      "113\t46\t0.0\t2\t44 2\n",
      "114\t53\t0.0\t2\t50 2 1\n",
      "115\t122\t0.0\t2\t119 2 1\n",
      "116\t212\t0.0\t2\t206 6\n",
      "117\t205\t0.0\t2\t199 5 1\n",
      "118\t345\t0.0\t2\t336 8 1\n",
      "119\t318\t0.0\t2\t302 15 1\n",
      "120\t268\t0.0\t2\t260 7 1\n",
      "121\t138\t0.0\t2\t134 4\n",
      "122\t160\t0.0\t2\t151 9\n",
      "123\t106\t0.0\t2\t101 3 2\n",
      "124\t93\t0.0\t2\t88 3 2\n",
      "125\t165\t0.0\t2\t161 4\n",
      "126\t114\t0.0\t2\t114\n",
      "127\t212\t0.0\t2\t208 4\n",
      "128\t186\t0.0\t2\t181 3 2\n",
      "129\t471\t0.0\t2\t466 4 1\n",
      "130\t2264\t0.0\t2\t2212 47 5\n",
      "131\t1185\t0.0\t2\t1162 20 3\n",
      "132\t663\t0.0\t2\t651 11 1\n",
      "133\t285\t0.0\t2\t279 5 1\n",
      "134\t471\t0.0\t2\t463 8\n",
      "135\t68\t0.0\t2\t63 5\n",
      "136\t50\t0.0\t2\t48 2\n",
      "137\t75\t0.0\t2\t73 2\n",
      "138\t40\t0.0\t2\t36 4\n",
      "139\t79\t0.0\t2\t75 3 1\n",
      "140\t220\t0.0\t2\t207 13\n",
      "141\t243\t0.0\t2\t49 117 77\n",
      "142\t3\t0.0\t2\t1 2\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n"
     ]
    }
   ],
   "source": [
    "log_file = filepath+'log_5RACE_cutadapt_3end.txt'\n",
    "###\n",
    "logs = open(log_file, 'w')\n",
    "logs.close()\n",
    "###\n",
    "\n",
    "for f in files:\n",
    "    print f\n",
    "    ###\n",
    "    cutadapt_inputfile = fastq_path+f+cutadapt_output_extension\n",
    "    cutadapt_outputfile = fastq_path+f+cutadapt_output_extension_3end\n",
    "\n",
    "    command = 'cutadapt -a %s %s -o %s %s >> %s' % (\n",
    "        cutadapt_adapter_3end,\n",
    "        cutadapt_option_3end, \n",
    "        cutadapt_outputfile,\n",
    "        cutadapt_inputfile,\n",
    "        log_file\n",
    "        )\n",
    "    \n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n\\n'+f+'\\n')\n",
    "    logs.close()\n",
    "    \n",
    "    os.system(command)\n",
    "    ###\n",
    "\n",
    "print \n",
    "for line in open(log_file, 'r'):\n",
    "    print line[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ks222\n",
      "ks223\n",
      "ks224\n",
      "ks225\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "ks222\n",
      "bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 --un alignments/StallReporter_secM/ks222_nomatch.fastq --max alignments/StallReporter_secM/ks222_multi.fastq --al alignments/StallReporter_secM/ks222_match.fastq StallReporters/StallReporter_secM fastq_5RACE/ks222-ctadpt_ctadpt3.fastq alignments/StallReporter_secM/ks222_StallReporter_secM_match.SAM 1>> log_5RACE_Bowtie_StallReporter_secM.txt 2>> log_5RACE_Bowtie_StallReporter_secM.txt\n",
      "# reads processed: 29299\n",
      "# reads with at least one reported alignment: 25177 (85.93%)\n",
      "# reads that failed to align: 4122 (14.07%)\n",
      "Reported 25184 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks223\n",
      "bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 --un alignments/StallReporter_secM/ks223_nomatch.fastq --max alignments/StallReporter_secM/ks223_multi.fastq --al alignments/StallReporter_secM/ks223_match.fastq StallReporters/StallReporter_secM fastq_5RACE/ks223-ctadpt_ctadpt3.fastq alignments/StallReporter_secM/ks223_StallReporter_secM_match.SAM 1>> log_5RACE_Bowtie_StallReporter_secM.txt 2>> log_5RACE_Bowtie_StallReporter_secM.txt\n",
      "# reads processed: 25408\n",
      "# reads with at least one reported alignment: 22067 (86.85%)\n",
      "# reads that failed to align: 3341 (13.15%)\n",
      "Reported 22070 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks224\n",
      "bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 --un alignments/StallReporter_secM/ks224_nomatch.fastq --max alignments/StallReporter_secM/ks224_multi.fastq --al alignments/StallReporter_secM/ks224_match.fastq StallReporters/StallReporter_secM fastq_5RACE/ks224-ctadpt_ctadpt3.fastq alignments/StallReporter_secM/ks224_StallReporter_secM_match.SAM 1>> log_5RACE_Bowtie_StallReporter_secM.txt 2>> log_5RACE_Bowtie_StallReporter_secM.txt\n",
      "# reads processed: 31372\n",
      "# reads with at least one reported alignment: 26380 (84.09%)\n",
      "# reads that failed to align: 4992 (15.91%)\n",
      "Reported 26385 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks225\n",
      "bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 --un alignments/StallReporter_secM/ks225_nomatch.fastq --max alignments/StallReporter_secM/ks225_multi.fastq --al alignments/StallReporter_secM/ks225_match.fastq StallReporters/StallReporter_secM fastq_5RACE/ks225-ctadpt_ctadpt3.fastq alignments/StallReporter_secM/ks225_StallReporter_secM_match.SAM 1>> log_5RACE_Bowtie_StallReporter_secM.txt 2>> log_5RACE_Bowtie_StallReporter_secM.txt\n",
      "# reads processed: 34413\n",
      "# reads with at least one reported alignment: 29363 (85.33%)\n",
      "# reads that failed to align: 5050 (14.67%)\n",
      "Reported 29365 alignments to 1 output stream(s)\n"
     ]
    }
   ],
   "source": [
    "######################################################\n",
    "\n",
    "files_reporter = ['ks'+str(x) for x in range(222,226)]\n",
    "stall_reporter = 'secM'\n",
    "\n",
    "######################################################\n",
    "\n",
    "###\n",
    "path_stallreporter = outpath + 'alignments/StallReporter_'+stall_reporter+'/'\n",
    "if not os.path.exists(outpath + 'alignments/'):\n",
    "    os.makedirs(outpath + 'alignments/')\n",
    "if not os.path.exists(path_stallreporter):\n",
    "    os.makedirs(path_stallreporter)\n",
    "###\n",
    "log_file = filepath+'log_5RACE_Bowtie_StallReporter_'+stall_reporter+'.txt'\n",
    "logs = open(log_file, 'w')\n",
    "logs.close()\n",
    "###     \n",
    "        \n",
    "for fname in files_reporter:\n",
    "    print fname\n",
    "\n",
    "    bowtie_command = 'bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 '\\\n",
    "        '--un %s%s --max %s%s --al %s%s %s %s%s %s%s 1>> %s 2>> %s'\n",
    "     \n",
    "    bowtie_stallreporter = bowtie_command %(\n",
    "        path_stallreporter,\n",
    "        fname+'_nomatch.fastq',\n",
    "        path_stallreporter,\n",
    "        fname+'_multi.fastq',\n",
    "        path_stallreporter,\n",
    "        fname+'_match.fastq',\n",
    "        fasta_path+'StallReporter_'+stall_reporter,\n",
    "        inpath,\n",
    "        fname + bowtie_input_extension,\n",
    "        path_stallreporter,\n",
    "        fname + '_StallReporter_'+stall_reporter+'_match.SAM',\n",
    "        log_file,\n",
    "        log_file\n",
    "        )\n",
    "    ###\n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n\\n'+fname+'\\n'+bowtie_stallreporter+'\\n')\n",
    "    logs.close()\n",
    "    ###\n",
    "    os.system(bowtie_stallreporter)           \n",
    "\n",
    "print \n",
    "print \n",
    "for line in open(log_file, 'r'):\n",
    "    print line[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fasta = fasta_path+'StallReporter_'+stall_reporter+'.fasta'\n",
    "\n",
    "out_path=filepath+'density_StallReporter/'\n",
    "if not os.path.exists(out_path):\n",
    "    os.makedirs(out_path)\n",
    "\n",
    "for f in files_reporter:\n",
    "    samgenmain_filename=filepath+'alignments/StallReporter_'+stall_reporter+'/'+f+'_StallReporter_'+stall_reporter+'_match.SAM'\n",
    "    f_samgenmain=open(samgenmain_filename)\n",
    "    samgenmain=csv.reader(f_samgenmain,delimiter='\\t')\n",
    "\n",
    "    three_mapped=five_end_mapping_stallreporter(samgenmain,fasta)\n",
    "    ###\n",
    "    pickle_readcounts = out_path+f+stall_reporter+mapping+'_readcounts.pickle'\n",
    "    with open(pickle_readcounts, 'wb') as readcounts_f:\n",
    "        pickle.dump(three_mapped,readcounts_f)\n",
    "    ###\n",
    "    three_mapped_rpm = deepcopy(three_mapped)\n",
    "    readcounts_to_rpm(three_mapped_rpm[0])\n",
    "    #readcounts_to_rpm(three_mapped_rpm[1])\n",
    "    pickle_rpm = out_path+f+stall_reporter+mapping+'_rpm.pickle'\n",
    "    with open(pickle_rpm, 'wb') as rpm_f:\n",
    "        pickle.dump(three_mapped_rpm,rpm_f)\n",
    "\n",
    "    density_path = out_path+f+stall_reporter+mapping\n",
    "    writecountsf(three_mapped_rpm[0],density_path+\"_plus_\")\n",
    "    countstowig(three_mapped_rpm[0],density_path+\"_plus\")\n",
    "    writecountsf(three_mapped_rpm[1],density_path+\"_minus_\")\n",
    "    countstowig(three_mapped_rpm[1],density_path+\"_minus\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ks230\n",
      "ks231\n",
      "ks232\n",
      "ks233\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "ks230\n",
      "bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 --un alignments/StallReporter_EPstop/ks230_nomatch.fastq --max alignments/StallReporter_EPstop/ks230_multi.fastq --al alignments/StallReporter_EPstop/ks230_match.fastq StallReporters/StallReporter_EPstop fastq_5RACE/ks230-ctadpt_ctadpt3.fastq alignments/StallReporter_EPstop/ks230_StallReporter_EPstop_match.SAM 1>> log_5RACE_Bowtie_StallReporter_EPstop.txt 2>> log_5RACE_Bowtie_StallReporter_EPstop.txt\n",
      "# reads processed: 31562\n",
      "# reads with at least one reported alignment: 25750 (81.59%)\n",
      "# reads that failed to align: 5812 (18.41%)\n",
      "Reported 25751 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks231\n",
      "bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 --un alignments/StallReporter_EPstop/ks231_nomatch.fastq --max alignments/StallReporter_EPstop/ks231_multi.fastq --al alignments/StallReporter_EPstop/ks231_match.fastq StallReporters/StallReporter_EPstop fastq_5RACE/ks231-ctadpt_ctadpt3.fastq alignments/StallReporter_EPstop/ks231_StallReporter_EPstop_match.SAM 1>> log_5RACE_Bowtie_StallReporter_EPstop.txt 2>> log_5RACE_Bowtie_StallReporter_EPstop.txt\n",
      "# reads processed: 33207\n",
      "# reads with at least one reported alignment: 27091 (81.58%)\n",
      "# reads that failed to align: 6116 (18.42%)\n",
      "Reported 27094 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks232\n",
      "bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 --un alignments/StallReporter_EPstop/ks232_nomatch.fastq --max alignments/StallReporter_EPstop/ks232_multi.fastq --al alignments/StallReporter_EPstop/ks232_match.fastq StallReporters/StallReporter_EPstop fastq_5RACE/ks232-ctadpt_ctadpt3.fastq alignments/StallReporter_EPstop/ks232_StallReporter_EPstop_match.SAM 1>> log_5RACE_Bowtie_StallReporter_EPstop.txt 2>> log_5RACE_Bowtie_StallReporter_EPstop.txt\n",
      "# reads processed: 32757\n",
      "# reads with at least one reported alignment: 26506 (80.92%)\n",
      "# reads that failed to align: 6251 (19.08%)\n",
      "Reported 26509 alignments to 1 output stream(s)\n",
      "\n",
      "\n",
      "ks233\n",
      "bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 --un alignments/StallReporter_EPstop/ks233_nomatch.fastq --max alignments/StallReporter_EPstop/ks233_multi.fastq --al alignments/StallReporter_EPstop/ks233_match.fastq StallReporters/StallReporter_EPstop fastq_5RACE/ks233-ctadpt_ctadpt3.fastq alignments/StallReporter_EPstop/ks233_StallReporter_EPstop_match.SAM 1>> log_5RACE_Bowtie_StallReporter_EPstop.txt 2>> log_5RACE_Bowtie_StallReporter_EPstop.txt\n",
      "# reads processed: 36484\n",
      "# reads with at least one reported alignment: 29479 (80.80%)\n",
      "# reads that failed to align: 7004 (19.20%)\n",
      "# reads with alignments suppressed due to -m: 1 (0.00%)\n",
      "Reported 29481 alignments to 1 output stream(s)\n"
     ]
    }
   ],
   "source": [
    "######################################################\n",
    "\n",
    "files_reporter = ['ks'+str(x) for x in range(230,234)]\n",
    "stall_reporter = 'EPstop'\n",
    "\n",
    "######################################################\n",
    "\n",
    "###\n",
    "path_stallreporter = outpath + 'alignments/StallReporter_'+stall_reporter+'/'\n",
    "if not os.path.exists(outpath + 'alignments/'):\n",
    "    os.makedirs(outpath + 'alignments/')\n",
    "if not os.path.exists(path_stallreporter):\n",
    "    os.makedirs(path_stallreporter)\n",
    "###\n",
    "log_file = filepath+'log_5RACE_Bowtie_StallReporter_'+stall_reporter+'.txt'\n",
    "logs = open(log_file, 'w')\n",
    "logs.close()\n",
    "###     \n",
    "        \n",
    "for fname in files_reporter:\n",
    "    print fname\n",
    "\n",
    "    bowtie_command = 'bowtie -v 2 -y -m 3 -a --best --strata -S -p 30 '\\\n",
    "        '--un %s%s --max %s%s --al %s%s %s %s%s %s%s 1>> %s 2>> %s'\n",
    "     \n",
    "    bowtie_stallreporter = bowtie_command %(\n",
    "        path_stallreporter,\n",
    "        fname+'_nomatch.fastq',\n",
    "        path_stallreporter,\n",
    "        fname+'_multi.fastq',\n",
    "        path_stallreporter,\n",
    "        fname+'_match.fastq',\n",
    "        fasta_path+'StallReporter_'+stall_reporter,\n",
    "        inpath,\n",
    "        fname + bowtie_input_extension,\n",
    "        path_stallreporter,\n",
    "        fname + '_StallReporter_'+stall_reporter+'_match.SAM',\n",
    "        log_file,\n",
    "        log_file\n",
    "        )\n",
    "    ###\n",
    "    logs = open(log_file, 'a+')\n",
    "    logs.write('\\n\\n'+fname+'\\n'+bowtie_stallreporter+'\\n')\n",
    "    logs.close()\n",
    "    ###\n",
    "    os.system(bowtie_stallreporter)           \n",
    "\n",
    "print \n",
    "print \n",
    "for line in open(log_file, 'r'):\n",
    "    print line[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fasta = fasta_path+'StallReporter_'+stall_reporter+'.fasta'\n",
    "\n",
    "out_path=filepath+'density_StallReporter/'\n",
    "if not os.path.exists(out_path):\n",
    "    os.makedirs(out_path)\n",
    "\n",
    "for f in files_reporter:\n",
    "    samgenmain_filename=filepath+'alignments/StallReporter_'+stall_reporter+'/'+f+'_StallReporter_'+stall_reporter+'_match.SAM'\n",
    "    f_samgenmain=open(samgenmain_filename)\n",
    "    samgenmain=csv.reader(f_samgenmain,delimiter='\\t')\n",
    "\n",
    "    three_mapped=five_end_mapping_stallreporter(samgenmain,fasta)\n",
    "    ###\n",
    "    pickle_readcounts = out_path+f+stall_reporter+mapping+'_readcounts.pickle'\n",
    "    with open(pickle_readcounts, 'wb') as readcounts_f:\n",
    "        pickle.dump(three_mapped,readcounts_f)\n",
    "    ###\n",
    "    three_mapped_rpm = deepcopy(three_mapped)\n",
    "    readcounts_to_rpm(three_mapped_rpm[0])\n",
    "    #readcounts_to_rpm(three_mapped_rpm[1])\n",
    "    pickle_rpm = out_path+f+stall_reporter+mapping+'_rpm.pickle'\n",
    "    with open(pickle_rpm, 'wb') as rpm_f:\n",
    "        pickle.dump(three_mapped_rpm,rpm_f)\n",
    "\n",
    "    density_path = out_path+f+stall_reporter+mapping\n",
    "    writecountsf(three_mapped_rpm[0],density_path+\"_plus_\")\n",
    "    countstowig(three_mapped_rpm[0],density_path+\"_plus\")\n",
    "    writecountsf(three_mapped_rpm[1],density_path+\"_minus_\")\n",
    "    countstowig(three_mapped_rpm[1],density_path+\"_minus\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
